%!TEX TS-program = xelatex
\documentclass[11pt]{article}

\usepackage[english]{babel}

\usepackage{amsmath,amssymb,amsfonts}
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
\usepackage{stix2}
\usepackage[scaled]{helvet}
\usepackage[scaled]{inconsolata}

\usepackage{lastpage}

\usepackage{setspace}

\usepackage{ccicons}

\usepackage[hang,flushmargin]{footmisc}

\usepackage{geometry}

\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}

\usepackage{fancyhdr}
\renewcommand{\headrulewidth}{0pt}\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}

\makeatletter
\newcounter{tableno}
\newenvironment{tablenos:no-prefix-table-caption}{
  \caption@ifcompatibility{}{
    \let\oldthetable\thetable
    \let\oldtheHtable\theHtable
    \renewcommand{\thetable}{tableno:\thetableno}
    \renewcommand{\theHtable}{tableno:\thetableno}
    \stepcounter{tableno}
    \captionsetup{labelformat=empty}
  }
}{
  \caption@ifcompatibility{}{
    \captionsetup{labelformat=default}
    \let\thetable\oldthetable
    \let\theHtable\oldtheHtable
    \addtocounter{table}{-1}
  }
}
\makeatother

\usepackage{array}
\newcommand{\PreserveBackslash}[1]{\let\temp=\\#1\let\\=\temp}
\let\PBS=\PreserveBackslash

\usepackage[breaklinks=true]{hyperref}
\hypersetup{colorlinks,%
citecolor=blue,%
filecolor=blue,%
linkcolor=blue,%
urlcolor=blue}
\usepackage{url}

\usepackage{caption}
\setcounter{secnumdepth}{0}
\usepackage{cleveref}

\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth
\else\Gin@nat@width\fi}
\makeatother
\let\Oldincludegraphics\includegraphics
\renewcommand{\includegraphics}[1]{\Oldincludegraphics[width=\maxwidth]{#1}}

\usepackage{longtable}
\usepackage{booktabs}

\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}

\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newlength{\csllabelwidth}
\setlength{\csllabelwidth}{3em}
\newenvironment{CSLReferences}[3] % #1 hanging-ident, #2 entry spacing
 {% don't indent paragraphs
  \setlength{\parindent}{0pt}
  % turn on hanging indent if param 1 is 1
  \ifodd #1 \everypar{\setlength{\hangindent}{\cslhangindent}}\ignorespaces\fi
  % set entry spacing
  \ifnum #2 > 0
  \setlength{\parskip}{#2\baselineskip}
  \fi
 }%
 {}
\usepackage{calc} % for \widthof, \maxof
\newcommand{\CSLBlock}[1]{#1\hfill\break}
\newcommand{\CSLLeftMargin}[1]{\parbox[t]{\maxof{\widthof{#1}}{\csllabelwidth}}{#1}}
\newcommand{\CSLRightInline}[1]{\parbox[t]{\linewidth}{#1}}
\newcommand{\CSLIndent}[1]{\hspace{\cslhangindent}#1}\geometry{verbose,letterpaper,tmargin=2.2cm,bmargin=2.2cm,lmargin=2.2cm,rmargin=2.2cm}

\usepackage{lineno}
\usepackage[nolists,noheads]{endfloat}

\pagestyle{plain}

\tolerance=1
\emergencystretch=\maxdimen
\hyphenpenalty=10000
\hbadness=10000

\doublespacing

\fancypagestyle{normal}
{
  \fancyhf{}
  \fancyfoot[R]{\footnotesize\sffamily\thepage\ of \pageref*{LastPage}}
}
\begin{document}
\raggedright
\thispagestyle{empty}
{\Large\bfseries\sffamily Predicting metawebs: graph embeddings can help
alleviate spatial data deficiencies}
\vskip 5em

%
\href{https://orcid.org/0000-0001-6067-1349}{Tanya\,Strydom}%
%
\,\textsuperscript{1,2,‡}\quad %
\href{https://orcid.org/0000-0002-0735-5184}{Timothée\,Poisot}%
%
\,\textsuperscript{1,2,‡}

\textsuperscript{1}\,Département de Sciences Biologiques, Université de
Montréal, Montréal, Canada\quad \textsuperscript{2}\,Quebec Centre for
Biodiversity Science, Montréal, Canada

\textsuperscript{‡}\,These authors contributed equally to the work\\

\textbf{Correspondance to:}\\
Timothée Poisot --- \texttt{timothee.poisot@umontreal.ca}\\

\vfill
This work is released by its authors under a CC-BY 4.0 license\hfill\ccby\\
Last revision: \emph{\today}

\clearpage
\thispagestyle{empty}

\vfill

\begin{enumerate}
    \item Metawebs, i.e.~networks of potential interactions within a
species pool, are a powerful abstraction to understand how large-scales
species interaction networks are structured.%
    \item Because metawebs are typically expressed at large spatial and
taxonomic scales, assembling them is a tedious and costly process;
predictive methods can help circumvent the limitations in data
deficiencies, by providing `draft' metawebs.%
    \item One way to improve the predictive ability is to maximize the
information used for prediction, by using graph embeddings rather than
the list of species interactions. Graph embedding is an emerging field
in machine learning that holds great potential for ecoloigcal problems.%
    \item In this perspective, we outline how the challenges associated
with infering metawebs line-up with the advantages of graph embeddings;
furthermore, because metawebs are inherently spatial objects, we discuss
how the choice of the species pool has consequences on the reconstructed
network, but also embeds hypotheses about which human-made boundaries
are ecologically meaningful.%
\end{enumerate}


\vfill

\clearpage
\linenumbers
\pagestyle{normal}

Having a general solution for inferring \emph{potential} interactions
(despite the unavailability of interaction data) could be the catalyst
for significant breakthroughs in our ability to start thinking about
species interaction networks over large spatial scales. In a recent
overview of the field of ecological network prediction, Strydom et al.
(2021) identified two challenges of interest to the prediction of
interactions at large scales. First, there is a relative scarcity of
relevant data in most places globally -- paradoxically, this restricts
our ability to infer interactions to locations where inference is
perhaps the least required; second, accurate predictions often demand
accurate predictors, and the lack of methods that can leverage small
amount of data is a serious impediment to our predictive ability
globally.

Following the definition of Dunne (2006), a metaweb is a network
analogue to the regional species pool; specifically, it is an inventory
of all \emph{potential} interactions between species from a spatially
delimited area (and so captures the \(\gamma\) diversity of
interactions). The metaweb is, therefore, \emph{not} a prediction of the
food web at a specific locale within the spatial area it covers, and
will have a different structure (notably by having a larger connectance;
see \emph{e.g.} Wood et al., 2015). These local food webs (which
captures the \(\alpha\) diversity of interactions) are a subset of the
metaweb's species and interactions, and have been called ``metaweb
realizations'' (Poisot et al., 2015). Differences between local food web
and their metaweb are due to chance, species abundance and
co-occurrence, local environmental conditions, and local distribution of
functional traits, among others.

Because the metaweb represents the joint effect of functional,
phylogenetic, and macroecological processes (Morales-Castilla et al.,
2015), it holds valuable ecological information. Specifically, it is the
``upper bounds'' on what the composition of the local networks can be
(see \emph{e.g.} McLeod et al., 2021). These local networks, in turn,
can be reconstructed given appropriate knowledge of local species
composition, providing information on structure of food webs at finer
spatial scales. This has been done for example for
tree-galler-parasitoid systems (Gravel et al., 2018), fish trophic
interactions (Albouy et al., 2019), tetrapod trophic interactions
(O'Connor et al., 2020), and crop-pest networks (Grünig et al., 2020).
Whereas the original metaweb definition, and indeed most past uses of
metawebs, was based on the presence/absence of interactions, we focus on
\emph{probabilistic} metawebs where interactions are represented as the
chance of success of a Bernoulli trial (see \emph{e.g.} Poisot et al.,
2016); therefore, not only does our method recommend interactions that
may exist, it gives each interaction a score, allowing us to properly
weigh them.

\hypertarget{the-metaweb-is-an-inherently-probabilistic-object}{%
\section{The metaweb is an inherently probabilistic
object}\label{the-metaweb-is-an-inherently-probabilistic-object}}

Yet, owing to the inherent plasticity of interactions, there have been
documented instances of food webs undergoing rapid collapse/recovery
cycles over short periods of time (Pedersen et al., 2017). The embedding
of a network, in a sense, embeds its macro-evolutionary history,
especially as RDPG captures ecological signal (Dalla Riva \& Stouffer,
2016); at this point, it is important to recall that a metaweb is
intended as a catalogue of all potential interactions, which should then
be filtered (Morales-Castilla et al., 2015). In practice (and in this
instance) the reconstructed metaweb will predict interactions that are
plausible based on the species' evolutionary history, however some
interactions would/would not be realized due to human impact.

Dallas et al. (2017) suggested that most links in ecological networks
may be cryptic, \emph{i.e.} uncommon or otherwise hard to observe. This
argument essentially echoes Jordano (2016): the sampling of ecological
interactions is difficult because it requires first the joint
observation of two species, and then the observation of their
interaction. In addition, it is generally expected that weak or rare
links would be more common in networks (Csermely, 2004), compared to
strong, persistent links; this is notably the case in food chains,
wherein many weaker links are key to the stability of a system (Neutel
et al., 2002). In the light of these observations, the results in
fig.~\ref{fig:inflation} are not particularly surprising: we expect to
see a surge in these low-probability interactions under a model that has
a good predictive accuracy. Because the predictions we generate are by
design probabilistic, then one can weigh these rare links appropriately.
In a sense, that most ecological interactions are elusive can call for a
slightly different approach to sampling: once the common interactions
are documented, the effort required in documenting each rare interaction
may increase exponentially. Recent proposals suggest that machine
learning algorithms, in these situations, can act as data generators
(Hoffmann et al., 2019): in this perspective, high quality observational
data can be supplemented with synthetic data coming from predictive
models, which increases the volume of information available for
inference. Indeed, Strydom et al. (2021) suggested that knowing the
metaweb may render the prediction of local networks easier, because it
fixes an ``upper bound'' on which interactions can exist; indeed, with a
probabilistic metaweb, we can consider that the metaweb represents an
aggregation of informative priors on the interactions.

\hypertarget{graph-embedding-offers-promises-for-the-inference-of-potential-interactions}{%
\section{Graph embedding offers promises for the inference of potential
interactions}\label{graph-embedding-offers-promises-for-the-inference-of-potential-interactions}}

Graph embedding is a varied family of machine learning techniques aiming
to transform nodes and edges into a vector space, usually of a lower
dimension, whilst maximally retaining key properties of the graph (Yan
et al., 2005). Ecological networks are an interesting candidate for the
widespread application of embeddings, as they tend to posess a shared
sstructural backbone (Mora et al., 2018), which hints at structural
invariants that can be revealed a lower dimensions. Indeed, previous
work by Eklöf et al. (2013) suggests that food webs are inherently
low-dimensional objects, and can be adequately represented with less
than ten dimensions. Simulation results by Botella et al. (2022) suggest
that there is no best method to identify architectural similarities
between networks, and that multiple approaches need to be tested and
compared to the network descriptor of interest.

But the popularity of graph embedding techniques in machine learning is
rather more intuitive than the search for structural invariants: while
graphs are discrete objects, machine learning techniques tend to handle
continuous data better. Therefore, bringing a discrete graph into a
continuous vector space opens up a broader variety of predictive
algorithms.

\hypertarget{tbl:methods}{}
\begin{longtable}[]{@{}llll@{}}
\caption{\label{tbl:methods}Overview of some common graph embedding
approaches, by time of publication, alongside examples of their use in
the prediction of species interactions. Surprisingly, these methods have
not yet been used routinely to predict species interactions; most of the
examples we identified were either statistical associations, or
analogues to joint species distribution models. \(^a\): statistical
interactions; \(^b\): joint-SDM-like approach.}\tabularnewline
\toprule
\begin{minipage}[b]{0.11\columnwidth}\raggedright
Method\strut
\end{minipage} & \begin{minipage}[b]{0.32\columnwidth}\raggedright
Embedding approach\strut
\end{minipage} & \begin{minipage}[b]{0.18\columnwidth}\raggedright
Reference\strut
\end{minipage} & \begin{minipage}[b]{0.28\columnwidth}\raggedright
Species interactions application\strut
\end{minipage}\tabularnewline
\midrule
\endfirsthead
\toprule
\begin{minipage}[b]{0.11\columnwidth}\raggedright
Method\strut
\end{minipage} & \begin{minipage}[b]{0.32\columnwidth}\raggedright
Embedding approach\strut
\end{minipage} & \begin{minipage}[b]{0.18\columnwidth}\raggedright
Reference\strut
\end{minipage} & \begin{minipage}[b]{0.28\columnwidth}\raggedright
Species interactions application\strut
\end{minipage}\tabularnewline
\midrule
\endhead
\begin{minipage}[t]{0.11\columnwidth}\raggedright
RDPG\strut
\end{minipage} & \begin{minipage}[t]{0.32\columnwidth}\raggedright
graphs through SVD\strut
\end{minipage} & \begin{minipage}[t]{0.18\columnwidth}\raggedright
Young \& Scheinerman (2007)\strut
\end{minipage} & \begin{minipage}[t]{0.28\columnwidth}\raggedright
Poisot et al. (2021)\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.11\columnwidth}\raggedright
tSNE\strut
\end{minipage} & \begin{minipage}[t]{0.32\columnwidth}\raggedright
nodes through statistical divergence\strut
\end{minipage} & \begin{minipage}[t]{0.18\columnwidth}\raggedright
Hinton \& Roweis (2002)\strut
\end{minipage} & \begin{minipage}[t]{0.28\columnwidth}\raggedright
Cieslak et al. (2020) \(^a\)\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.11\columnwidth}\raggedright
DeepWalk\strut
\end{minipage} & \begin{minipage}[t]{0.32\columnwidth}\raggedright
graph walk\strut
\end{minipage} & \begin{minipage}[t]{0.18\columnwidth}\raggedright
Perozzi et al. (2014)\strut
\end{minipage} & \begin{minipage}[t]{0.28\columnwidth}\raggedright
Wardeh et al. (2021)\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.11\columnwidth}\raggedright
FastEmbed\strut
\end{minipage} & \begin{minipage}[t]{0.32\columnwidth}\raggedright
graph through PCA/SVD analogue\strut
\end{minipage} & \begin{minipage}[t]{0.18\columnwidth}\raggedright
Ramasamy \& Madhow (2015)\strut
\end{minipage} & \begin{minipage}[t]{0.28\columnwidth}\raggedright
\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.11\columnwidth}\raggedright
LINE\strut
\end{minipage} & \begin{minipage}[t]{0.32\columnwidth}\raggedright
nodes through statistical divergence\strut
\end{minipage} & \begin{minipage}[t]{0.18\columnwidth}\raggedright
Tang et al. (2015)\strut
\end{minipage} & \begin{minipage}[t]{0.28\columnwidth}\raggedright
\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.11\columnwidth}\raggedright
SDNE\strut
\end{minipage} & \begin{minipage}[t]{0.32\columnwidth}\raggedright
nodes through auto-encoding\strut
\end{minipage} & \begin{minipage}[t]{0.18\columnwidth}\raggedright
D. Wang et al. (2016)\strut
\end{minipage} & \begin{minipage}[t]{0.28\columnwidth}\raggedright
\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.11\columnwidth}\raggedright
node2vec\strut
\end{minipage} & \begin{minipage}[t]{0.32\columnwidth}\raggedright
node embedding\strut
\end{minipage} & \begin{minipage}[t]{0.18\columnwidth}\raggedright
Grover \& Leskovec (2016)\strut
\end{minipage} & \begin{minipage}[t]{0.28\columnwidth}\raggedright
\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.11\columnwidth}\raggedright
graph2vec\strut
\end{minipage} & \begin{minipage}[t]{0.32\columnwidth}\raggedright
sub-graph embedding\strut
\end{minipage} & \begin{minipage}[t]{0.18\columnwidth}\raggedright
Narayanan et al. (2017)\strut
\end{minipage} & \begin{minipage}[t]{0.28\columnwidth}\raggedright
\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.11\columnwidth}\raggedright
DMSE\strut
\end{minipage} & \begin{minipage}[t]{0.32\columnwidth}\raggedright
joint node embedding\strut
\end{minipage} & \begin{minipage}[t]{0.18\columnwidth}\raggedright
D. Chen et al. (2017)\strut
\end{minipage} & \begin{minipage}[t]{0.28\columnwidth}\raggedright
D. Chen et al. (2017) \(^b\)\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.11\columnwidth}\raggedright
HARP\strut
\end{minipage} & \begin{minipage}[t]{0.32\columnwidth}\raggedright
nodes through a meta-strategy\strut
\end{minipage} & \begin{minipage}[t]{0.18\columnwidth}\raggedright
H. Chen et al. (2017)\strut
\end{minipage} & \begin{minipage}[t]{0.28\columnwidth}\raggedright
\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.11\columnwidth}\raggedright
GraphKKE\strut
\end{minipage} & \begin{minipage}[t]{0.32\columnwidth}\raggedright
graph embedding\strut
\end{minipage} & \begin{minipage}[t]{0.18\columnwidth}\raggedright
Melnyk et al. (2020)\strut
\end{minipage} & \begin{minipage}[t]{0.28\columnwidth}\raggedright
Melnyk et al. (2020) \(^a\)\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.11\columnwidth}\raggedright
Joint methods\strut
\end{minipage} & \begin{minipage}[t]{0.32\columnwidth}\raggedright
multiple graphs\strut
\end{minipage} & \begin{minipage}[t]{0.18\columnwidth}\raggedright
S. Wang et al. (2021)\strut
\end{minipage} & \begin{minipage}[t]{0.28\columnwidth}\raggedright
\strut
\end{minipage}\tabularnewline
\bottomrule
\end{longtable}

\hypertarget{the-metaweb-embeds-hypotheses-about-which-spatial-boundaries-are-meaningful}{%
\section{The metaweb embeds hypotheses about which spatial boundaries
are
meaningful}\label{the-metaweb-embeds-hypotheses-about-which-spatial-boundaries-are-meaningful}}

As Herbert (1965) rightfully pointed out, ``{[}y{]}ou can't draw neat
lines around planet-wide problems''; in this regard, our approach (and
indeed, any inference of a metaweb at large scales) must contend with
several interesting and interwoven families of problems. The first is
the limit of the metaweb to embed and transfer. If the initial metaweb
is too narrow in scope, notably from a taxonomic point of view, the
chances of finding another area with enough related species to make a
reliable inference decreases; this would likely be indicated by large
confidence intervals during ancestral character estimation, but the lack
of well documented metawebs is currently preventing the development of
more concrete guidelines. The question of phylogenetic relatedness and
dispersal is notably true if the metaweb is assembled in an area with
mostly endemic species, and as with every predictive algorithm, there is
room for the application of our best ecological judgement. Conversely,
the metaweb should be reliably filled, which assumes that the \(S^2\)
interactions in a pool of \(S\) species have been examined, either
through literature surveys or expert elicitation. Supp. Mat. 1 provides
some guidance as to the type of sampling effort that should be
prioritized. While RDPG was able to maintain very high predictive power
when interactions were missing, the addition of false positive
interactions was immediately detected; this suggests that it may be
appropriate to err on the side of ``too many'' interactions when
constructing the initial metaweb to be transferred. The second series of
problems are related to determining which area should be used to infer
the new metaweb in, as this determines the species pool that must be
used.

In our application, we focused on the mammals of Canada. The upside of
this approach is that information at the country level is likely to be
required by policy makers and stakeholders for their biodiversity
assessment, as each country tends to set goals at the national level
(Buxton et al., 2021) for which quantitative instruments are designed
(Turak et al., 2017), with specific strategies often enacted at smaller
scales (Ray et al., 2021). And yet, we do not really have a satisfying
answer to the question of ``where does a food web stop?''; the current
most satisfying solutions involve examining the spatial consistency of
network area relationships (see \emph{e.g.} Galiana et al., 2018, 2019,
2021; \textbf{Fortin2021NetEco?}), which is of course impossible in the
absence of enough information about the network itself. This suggests
that an \emph{a posteriori} refinement of the results may be required,
based on a downscaling of the metaweb. The final family of problems
relates less to the availability of data or quantitative tools, and more
to the praxis of spatial ecology. Operating under the context of
national divisions, in large parts of the world, reflects nothing more
than the legacy of settler colonialism. Indeed, the use of ecological
data is not an apolitical act (Nost \& Goldstein, 2021), as data
infrastructures tend to be designed to answer questions within national
boundaries, and their use both draws upon and reinforces territorial
statecraft; as per Machen \& Nost (2021), this is particularly true when
the output of ``algorithmic thinking'' (\emph{e.g.} relying on machine
learning to generate knowledge) can be re-used for governance
(\emph{e.g.} enacting conservation decisions at the national scale). We
therefore recognize that methods such as we propose operate under the
framework that contributed to the ongoing biodiversity crisis (Adam,
2014), reinforced environmental injustice (Choudry, 2013; Domínguez \&
Luoma, 2020), and on Turtle Island especially, should be replaced by
Indigenous principles of land management (Eichhorn et al., 2019; No'kmaq
et al., 2021). As we see AI/ML being increasingly mobilized to generate
knowledge that is lacking for conservation decisions (\emph{e.g.} Lamba
et al., 2019; Mosebo Fernandes et al., 2020), our discussion of these
tools need to go beyond the technical, and into the governance
consequences they can have.

\textbf{Acknowledgements:} We acknowledge that this study was conducted
on land within the traditional unceded territory of the Saint Lawrence
Iroquoian, Anishinabewaki, Mohawk, Huron-Wendat, and Omàmiwininiwak
nations. TP, TS, DC, and LP received funding from the Canadian Institue
for Ecology \& Evolution. FB is funded by the Institute for Data
Valorization (IVADO). TS, SB, and TP are funded by a donation from the
Courtois Foundation. CB was awarded a Mitacs Elevate Fellowship no.
IT12391, in partnership with fRI Research, and also acknowledges funding
from Alberta Innovates and the Forest Resources Improvement Association
of Alberta. M-JF acknowledges funding from NSERC Discovery Grant and
NSERC CRC. RR is funded by New Zealand's Biological Heritage Ngā Koiora
Tuku Iho National Science Challenge, administered by New Zealand
Ministry of Business, Innovation, and Employment. BM is funded by the
NSERC Alexander Graham Bell Canada Graduate Scholarship and the FRQNT
master's scholarship. LP acknowledges funding from NSERC Discovery Grant
(NSERC RGPIN-2019-05771). TP acknowledges financial support from NSERC
through the Discovery Grants and Discovery Accelerator Supplement
programs.

\hypertarget{references}{%
\section*{References}\label{references}}
\addcontentsline{toc}{section}{References}

\hypertarget{refs}{}
\begin{CSLReferences}{1}{0}
\leavevmode\hypertarget{ref-Adam2014EleTre}{}%
Adam, R. (2014). \emph{Elephant treaties: The Colonial legacy of the
biodiversity crisis}. UPNE.

\leavevmode\hypertarget{ref-Albouy2019MarFis}{}%
Albouy, C., Archambault, P., Appeltans, W., Araújo, M. B., Beauchesne,
D., Cazelles, K., Cirtwill, A. R., Fortin, M.-J., Galiana, N., Leroux,
S. J., Pellissier, L., Poisot, T., Stouffer, D. B., Wood, S. A., \&
Gravel, D. (2019). The marine fish food web is globally connected.
\emph{Nature Ecology \& Evolution}, \emph{3}(8, 8), 1153--1161.
\url{https://doi.org/10.1038/s41559-019-0950-y}

\leavevmode\hypertarget{ref-Botella2022AppGra}{}%
Botella, C., Dray, S., Matias, C., Miele, V., \& Thuiller, W. (2022). An
appraisal of graph embeddings for comparing trophic network
architectures. \emph{Methods in Ecology and Evolution}, \emph{13}(1),
203--216. \url{https://doi.org/10.1111/2041-210X.13738}

\leavevmode\hypertarget{ref-Buxton2021KeyInf}{}%
Buxton, R. T., Bennett, J. R., Reid, A. J., Shulman, C., Cooke, S. J.,
Francis, C. M., Nyboer, E. A., Pritchard, G., Binley, A. D., Avery-Gomm,
S., Ban, N. C., Beazley, K. F., Bennett, E., Blight, L. K., Bortolotti,
L. E., Camfield, A. F., Gadallah, F., Jacob, A. L., Naujokaitis-Lewis,
I., \ldots{} Smith, P. A. (2021). Key information needs to move from
knowledge to action for biodiversity conservation in Canada.
\emph{Biological Conservation}, \emph{256}, 108983.
\url{https://doi.org/10.1016/j.biocon.2021.108983}

\leavevmode\hypertarget{ref-Chen2017DeeMul}{}%
Chen, D., Xue, Y., Fink, D., Chen, S., \& Gomes, C. P. (2017).
\emph{Deep Multi-species Embedding}. 3639--3646.
\url{https://www.ijcai.org/proceedings/2017/509}

\leavevmode\hypertarget{ref-Chen2017HarHie}{}%
Chen, H., Perozzi, B., Hu, Y., \& Skiena, S. (2017, November 16).
\emph{HARP: Hierarchical Representation Learning for Networks}.
\url{http://arxiv.org/abs/1706.07845}

\leavevmode\hypertarget{ref-Choudry2013SavBio}{}%
Choudry, A. (2013). Saving biodiversity, for whom and for what?
Conservation NGOs, complicity, colonialism and conquest in an era of
capitalist globalization. In \emph{NGOization: Complicity,
contradictions and prospects} (pp. 24--44). Bloomsbury Publishing.

\leavevmode\hypertarget{ref-Cieslak2020TdiSto}{}%
Cieslak, M. C., Castelfranco, A. M., Roncalli, V., Lenz, P. H., \&
Hartline, D. K. (2020). T-Distributed Stochastic Neighbor Embedding
(t-SNE): A tool for eco-physiological transcriptomic analysis.
\emph{Marine Genomics}, \emph{51}, 100723.
\url{https://doi.org/10.1016/j.margen.2019.100723}

\leavevmode\hypertarget{ref-Csermely2004StrLin}{}%
Csermely, P. (2004). Strong links are important, but weak links
stabilize them. \emph{Trends in Biochemical Sciences}, \emph{29}(7),
331--334. \url{https://doi.org/10.1016/j.tibs.2004.05.004}

\leavevmode\hypertarget{ref-DallaRiva2016ExpEvo}{}%
Dalla Riva, G. V., \& Stouffer, D. B. (2016). Exploring the evolutionary
signature of food webs' backbones using functional traits. \emph{Oikos},
\emph{125}(4), 446--456. \url{https://doi.org/10.1111/oik.02305}

\leavevmode\hypertarget{ref-Dallas2017PreCry}{}%
Dallas, T., Park, A. W., \& Drake, J. M. (2017). Predicting cryptic
links in host-parasite networks. \emph{PLOS Computational Biology},
\emph{13}(5), e1005557.
\url{https://doi.org/10.1371/journal.pcbi.1005557}

\leavevmode\hypertarget{ref-Dominguez2020DecCon}{}%
Domínguez, L., \& Luoma, C. (2020). Decolonising Conservation Policy:
How Colonial Land and Conservation Ideologies Persist and Perpetuate
Indigenous Injustices at the Expense of the Environment. \emph{Land},
\emph{9}(3, 3), 65. \url{https://doi.org/10.3390/land9030065}

\leavevmode\hypertarget{ref-Dunne2006NetStr}{}%
Dunne, J. A. (2006). The Network Structure of Food Webs. In J. A. Dunne
\& M. Pascual (Eds.), \emph{Ecological networks: Linking structure and
dynamics} (pp. 27--86). Oxford University Press.

\leavevmode\hypertarget{ref-Eichhorn2019SteDec}{}%
Eichhorn, M. P., Baker, K., \& Griffiths, M. (2019). Steps towards
decolonising biogeography. \emph{Frontiers of Biogeography},
\emph{12}(1), 1--7. \url{https://doi.org/10.21425/F5FBG44795}

\leavevmode\hypertarget{ref-Eklof2013DimEco}{}%
Eklöf, A., Jacob, U., Kopp, J., Bosch, J., Castro-Urgal, R., Chacoff, N.
P., Dalsgaard, B., de Sassi, C., Galetti, M., Guimarães, P. R.,
Lomáscolo, S. B., Martín González, A. M., Pizo, M. A., Rader, R.,
Rodrigo, A., Tylianakis, J. M., Vázquez, D. P., \& Allesina, S. (2013).
The dimensionality of ecological networks. \emph{Ecology Letters},
\emph{16}(5), 577--583. \url{https://doi.org/10.1111/ele.12081}

\leavevmode\hypertarget{ref-Galiana2021SpaSca}{}%
Galiana, N., Barros, C., Braga, J., Ficetola, G. F., Maiorano, L.,
Thuiller, W., Montoya, J. M., \& Lurgi, M. (2021). The spatial scaling
of food web structure across European biogeographical regions.
\emph{Ecography}, \emph{n/a}(n/a).
\url{https://doi.org/10.1111/ecog.05229}

\leavevmode\hypertarget{ref-Galiana2019GeoVar}{}%
Galiana, N., Hawkins, B. A., \& Montoya, J. M. (2019). The geographical
variation of network structure is scale dependent: Understanding the
biotic specialization of host--parasitoid networks. \emph{Ecography},
\emph{42}(6), 1175--1187. \url{https://doi.org/10.1111/ecog.03684}

\leavevmode\hypertarget{ref-Galiana2018SpaSca}{}%
Galiana, N., Lurgi, M., Claramunt-López, B., Fortin, M.-J., Leroux, S.,
Cazelles, K., Gravel, D., \& Montoya, J. M. (2018). The spatial scaling
of species interaction networks. \emph{Nature Ecology \& Evolution},
\emph{2}(5), 782--790. \url{https://doi.org/10.1038/s41559-018-0517-3}

\leavevmode\hypertarget{ref-Gravel2018BriElt}{}%
Gravel, D., Baiser, B., Dunne, J. A., Kopelke, J.-P., Martinez, N. D.,
Nyman, T., Poisot, T., Stouffer, D. B., Tylianakis, J. M., Wood, S. A.,
\& Roslin, T. (2018). Bringing Elton and Grinnell together: A
quantitative framework to represent the biogeography of ecological
interaction networks. \emph{Ecography}, \emph{0}(0).
\url{https://doi.org/10.1111/ecog.04006}

\leavevmode\hypertarget{ref-Grover2016NodSca}{}%
Grover, A., \& Leskovec, J. (2016). Node2vec: Scalable Feature Learning
for Networks. \emph{Proceedings of the 22nd ACM SIGKDD International
Conference on Knowledge Discovery and Data Mining}, 855--864.
\url{https://doi.org/10.1145/2939672.2939754}

\leavevmode\hypertarget{ref-Grunig2020CroFor}{}%
Grünig, M., Mazzi, D., Calanca, P., Karger, D. N., \& Pellissier, L.
(2020). Crop and forest pest metawebs shift towards increased linkage
and suitability overlap under climate change. \emph{Communications
Biology}, \emph{3}(1, 1), 1--10.
\url{https://doi.org/10.1038/s42003-020-0962-9}

\leavevmode\hypertarget{ref-Herbert1965Dun}{}%
Herbert, F. (1965). \emph{Dune} (1st ed.). Chilton Book Company.

\leavevmode\hypertarget{ref-Hinton2002StoNei}{}%
Hinton, G., \& Roweis, S. T. (2002). Stochastic neighbor embedding.
\emph{NIPS}, \emph{15}, 833--840.

\leavevmode\hypertarget{ref-Hoffmann2019MacLea}{}%
Hoffmann, J., Bar-Sinai, Y., Lee, L. M., Andrejevic, J., Mishra, S.,
Rubinstein, S. M., \& Rycroft, C. H. (2019). Machine learning in a
data-limited regime: Augmenting experiments with synthetic data uncovers
order in crumpled sheets. \emph{Science Advances}, \emph{5}(4),
eaau6792. \url{https://doi.org/10.1126/sciadv.aau6792}

\leavevmode\hypertarget{ref-Jordano2016SamNet}{}%
Jordano, P. (2016). Sampling networks of ecological interactions.
\emph{Functional Ecology}, \emph{30}(12), 1883--1893.
\url{https://doi.org/10.1111/1365-2435.12763}

\leavevmode\hypertarget{ref-Lamba2019DeeLea}{}%
Lamba, A., Cassey, P., Segaran, R. R., \& Koh, L. P. (2019). Deep
learning for environmental conservation. \emph{Current Biology},
\emph{29}(19), R977--R982.
\url{https://doi.org/10.1016/j.cub.2019.08.016}

\leavevmode\hypertarget{ref-Machen2021ThiAlg}{}%
Machen, R., \& Nost, E. (2021). Thinking algorithmically: The making of
hegemonic knowledge in climate governance. \emph{Transactions of the
Institute of British Geographers}, \emph{46}(3), 555--569.
\url{https://doi.org/10.1111/tran.12441}

\leavevmode\hypertarget{ref-McLeod2021SamAsy}{}%
McLeod, A., Leroux, S. J., Gravel, D., Chu, C., Cirtwill, A. R., Fortin,
M.-J., Galiana, N., Poisot, T., \& Wood, S. A. (2021). Sampling and
asymptotic network properties of spatial multi-trophic networks.
\emph{Oikos}, \emph{n/a}(n/a). \url{https://doi.org/10.1111/oik.08650}

\leavevmode\hypertarget{ref-Melnyk2020GraGra}{}%
Melnyk, K., Klus, S., Montavon, G., \& Conrad, T. O. F. (2020).
GraphKKE: Graph Kernel Koopman embedding for human microbiome analysis.
\emph{Applied Network Science}, \emph{5}(1), 96.
\url{https://doi.org/10.1007/s41109-020-00339-2}

\leavevmode\hypertarget{ref-Mora2018IdeCom}{}%
Mora, B. B., Gravel, D., Gilarranz, L. J., Poisot, T., \& Stouffer, D.
B. (2018). Identifying a common backbone of interactions underlying food
webs from different ecosystems. \emph{Nature Communications},
\emph{9}(1), 2603. \url{https://doi.org/10.1038/s41467-018-05056-0}

\leavevmode\hypertarget{ref-Morales-Castilla2015InfBio}{}%
Morales-Castilla, I., Matias, M. G., Gravel, D., \& Araújo, M. B.
(2015). Inferring biotic interactions from proxies. \emph{Trends in
Ecology \& Evolution}, \emph{30}(6), 347--356.
\url{https://doi.org/10.1016/j.tree.2015.03.014}

\leavevmode\hypertarget{ref-MoseboFernandes2020MacLea}{}%
Mosebo Fernandes, A. C., Quintero Gonzalez, R., Lenihan-Clarke, M. A.,
Leslie Trotter, E. F., \& Jokar Arsanjani, J. (2020). Machine Learning
for Conservation Planning in a Changing Climate. \emph{Sustainability},
\emph{12}(18, 18), 7657. \url{https://doi.org/10.3390/su12187657}

\leavevmode\hypertarget{ref-Narayanan2017GraLea}{}%
Narayanan, A., Chandramohan, M., Venkatesan, R., Chen, L., Liu, Y., \&
Jaiswal, S. (2017, July 17). \emph{Graph2vec: Learning Distributed
Representations of Graphs}. \url{http://arxiv.org/abs/1707.05005}

\leavevmode\hypertarget{ref-Neutel2002StaRea}{}%
Neutel, A.-M., Heesterbeek, J. A. P., \& de Ruiter, P. C. (2002).
Stability in Real Food Webs: Weak Links in Long Loops. \emph{Science},
\emph{296}(5570), 1120--1123.
\url{https://doi.org/10.1126/science.1068326}

\leavevmode\hypertarget{ref-Nokmaq2021AwaSle}{}%
No'kmaq, M., Marshall, A., Beazley, K. F., Hum, J., joudry, shalan,
Papadopoulos, A., Pictou, S., Rabesca, J., Young, L., \& Zurba, M.
(2021). {``Awakening the sleeping giant''}: Re-Indigenization principles
for transforming biodiversity conservation in Canada and beyond.
\emph{FACETS}, \emph{6}(1), 839--869.

\leavevmode\hypertarget{ref-Nost2021PolEco}{}%
Nost, E., \& Goldstein, J. E. (2021). A political ecology of data.
\emph{Environment and Planning E: Nature and Space}, 25148486211043503.
\url{https://doi.org/10.1177/25148486211043503}

\leavevmode\hypertarget{ref-OConnor2020UnvFoo}{}%
O'Connor, L. M. J., Pollock, L. J., Braga, J., Ficetola, G. F.,
Maiorano, L., Martinez‐Almoyna, C., Montemaggiori, A., Ohlmann, M., \&
Thuiller, W. (2020). Unveiling the food webs of tetrapods across Europe
through the prism of the Eltonian niche. \emph{Journal of Biogeography},
\emph{47}(1), 181--192. \url{https://doi.org/10.1111/jbi.13773}

\leavevmode\hypertarget{ref-Pedersen2017SigCol}{}%
Pedersen, E. J., Thompson, P. L., Ball, R. A., Fortin, M.-J., Gouhier,
T. C., Link, H., Moritz, C., Nenzen, H., Stanley, R. R. E., Taranu, Z.
E., Gonzalez, A., Guichard, F., \& Pepin, P. (2017). Signatures of the
collapse and incipient recovery of an overexploited marine ecosystem.
\emph{Royal Society Open Science}, \emph{4}(7), 170215.
\url{https://doi.org/10.1098/rsos.170215}

\leavevmode\hypertarget{ref-Perozzi2014DeeOnl}{}%
Perozzi, B., Al-Rfou, R., \& Skiena, S. (2014). DeepWalk: Online
learning of social representations. \emph{Proceedings of the 20th ACM
SIGKDD International Conference on Knowledge Discovery and Data Mining},
701--710. \url{https://doi.org/10.1145/2623330.2623732}

\leavevmode\hypertarget{ref-Poisot2016StrPro}{}%
Poisot, T., Cirtwill, A. R., Cazelles, K., Gravel, D., Fortin, M.-J., \&
Stouffer, D. B. (2016). The structure of probabilistic networks.
\emph{Methods in Ecology and Evolution}, \emph{7}(3), 303--312.
\url{https://doi.org/10.1111/2041-210X.12468}

\leavevmode\hypertarget{ref-Poisot2021ImpMam}{}%
Poisot, T., Ouellet, M.-A., Mollentze, N., Farrell, M. J., Becker, D.
J., Albery, G. F., Gibb, R. J., Seifert, S. N., \& Carlson, C. J. (2021,
May 31). \emph{Imputing the mammalian virome with linear filtering and
singular value decomposition}. \url{http://arxiv.org/abs/2105.14973}

\leavevmode\hypertarget{ref-Poisot2015SpeWhy}{}%
Poisot, T., Stouffer, D. B., \& Gravel, D. (2015). Beyond species: Why
ecological interaction networks vary through space and time.
\emph{Oikos}, \emph{124}(3), 243--251.
\url{https://doi.org/10.1111/oik.01719}

\leavevmode\hypertarget{ref-Ramasamy2015ComSpe}{}%
Ramasamy, D., \& Madhow, U. (2015). Compressive spectral embedding:
Sidestepping the SVD. In C. Cortes, N. Lawrence, D. Lee, M. Sugiyama, \&
R. Garnett (Eds.), \emph{Advances in neural information processing
systems} (Vol. 28). Curran Associates, Inc.
\url{https://proceedings.neurips.cc/paper/2015/file/4f6ffe13a5d75b2d6a3923922b3922e5-Paper.pdf}

\leavevmode\hypertarget{ref-Ray2021BioCri}{}%
Ray, J. C., Grimm, J., \& Olive, A. (2021). The biodiversity crisis in
Canada: Failures and challenges of federal and sub-national strategic
and legal frameworks. \emph{FACETS}, \emph{6}, 1044--1068.
\url{https://doi.org/10.1139/facets-2020-0075}

\leavevmode\hypertarget{ref-Strydom2021RoaPre}{}%
Strydom, T., Catchen, M. D., Banville, F., Caron, D., Dansereau, G.,
Desjardins-Proulx, P., Forero-Muñoz, N. R., Higino, G., Mercier, B.,
Gonzalez, A., Gravel, D., Pollock, L., \& Poisot, T. (2021). A roadmap
towards predicting species interaction networks (across space and time).
\emph{Philosophical Transactions of the Royal Society B: Biological
Sciences}, \emph{376}(1837), 20210063.
\url{https://doi.org/10.1098/rstb.2021.0063}

\leavevmode\hypertarget{ref-Tang2015LinLar}{}%
Tang, J., Qu, M., Wang, M., Zhang, M., Yan, J., \& Mei, Q. (2015). LINE:
Large-scale Information Network Embedding. \emph{Proceedings of the 24th
International Conference on World Wide Web}, 1067--1077.
\url{https://doi.org/10.1145/2736277.2741093}

\leavevmode\hypertarget{ref-Turak2017UsiEss}{}%
Turak, E., Brazill-Boast, J., Cooney, T., Drielsma, M., DelaCruz, J.,
Dunkerley, G., Fernandez, M., Ferrier, S., Gill, M., Jones, H., Koen,
T., Leys, J., McGeoch, M., Mihoub, J.-B., Scanes, P., Schmeller, D., \&
Williams, K. (2017). Using the essential biodiversity variables
framework to measure biodiversity change at national scale.
\emph{Biological Conservation}, \emph{213}, 264--271.
\url{https://doi.org/10.1016/j.biocon.2016.08.019}

\leavevmode\hypertarget{ref-Wang2016StrDee}{}%
Wang, D., Cui, P., \& Zhu, W. (2016). Structural Deep Network Embedding.
\emph{Proceedings of the 22nd ACM SIGKDD International Conference on
Knowledge Discovery and Data Mining}, 1225--1234.
\url{https://doi.org/10.1145/2939672.2939753}

\leavevmode\hypertarget{ref-Wang2021JoiEmb}{}%
Wang, S., Arroyo, J., Vogelstein, J. T., \& Priebe, C. E. (2021). Joint
Embedding of Graphs. \emph{IEEE Transactions on Pattern Analysis and
Machine Intelligence}, \emph{43}(4), 1324--1336.
\url{https://doi.org/10.1109/TPAMI.2019.2948619}

\leavevmode\hypertarget{ref-Wardeh2021PreMam}{}%
Wardeh, M., Baylis, M., \& Blagrove, M. S. C. (2021). Predicting
mammalian hosts in which novel coronaviruses can be generated.
\emph{Nature Communications}, \emph{12}(1, 1), 780.
\url{https://doi.org/10.1038/s41467-021-21034-5}

\leavevmode\hypertarget{ref-Wood2015EffSpa}{}%
Wood, S. A., Russell, R., Hanson, D., Williams, R. J., \& Dunne, J. A.
(2015). Effects of spatial scale of sampling on food web structure.
\emph{Ecology and Evolution}, \emph{5}(17), 3769--3782.
\url{https://doi.org/10.1002/ece3.1640}

\leavevmode\hypertarget{ref-Yan2005GraEmb}{}%
Yan, S., Xu, D., Zhang, B., \& Zhang, H.-J. (2005). Graph embedding: A
general framework for dimensionality reduction. \emph{2005 IEEE Computer
Society Conference on Computer Vision and Pattern Recognition
(CVPR'05)}, \emph{2}, 830--837 vol. 2.
\url{https://doi.org/10.1109/CVPR.2005.170}

\leavevmode\hypertarget{ref-Young2007RanDot}{}%
Young, S. J., \& Scheinerman, E. R. (2007). Random Dot Product Graph
Models for Social Networks. In A. Bonato \& F. R. K. Chung (Eds.),
\emph{Algorithms and Models for the Web-Graph} (pp. 138--149). Springer.
\url{https://doi.org/10.1007/978-3-540-77004-6_11}

\end{CSLReferences}

\end{document}
