<!DOCTYPE html>
<html lang="" dir="ltr">

<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <title>Graph embedding and transfer learning can help predict
potential species interaction networks despite data limitations</title>
    <script src="https://hypothes.is/embed.js" async></script>
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/normalize/8.0.1/normalize.min.css"
    integrity="sha256-l85OmPOjvil/SOvVt3HnSSjzF1TUMyT9eV0c2BzEGzU=" crossorigin="anonymous" />
    <style>
        @import url('https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,300;0,400;0,500;1,300;1,400;1,500&display=swap');
        @import url('https://fonts.googleapis.com/css2?family=Roboto+Condensed:ital,wght@0,300;0,400;0,700;1,300;1,400;1,700&display=swap');
        @import url('https://fonts.googleapis.com/css2?family=Roboto+Mono:ital,wght@0,400;0,500;1,400;1,500&display=swap');

        :root {
            --citation: #1976D2;
            --structure: hsl(210, 61%, 43%);

            --light: #474747;
            --pale: #a5a5a5;
            --grey: #e0e0e0;
            --nav: #fdfdfd;

            --width: 820px;
            --inner-space: 1.2em;
        }

        body {
            font-family: "Roboto", sans-serif;
            margin: 0 0;
            padding: 0 0;
        }

        article>p {
            line-height: 135%;
        }

        article>h1 {
            padding-top: 3em;
        }

        a {
            text-decoration: none;
            color: #000;
            border-bottom: 1px solid var(--structure);
        }

        a:hover {
            font-weight: 500;
        }

        .citation {
            color: var(--citation);
        }

        .container {
            margin: 0em auto;
            max-width: var(--width);
            padding: var(--inner-space);
        }

        header>.container {
            margin-top: 4em;
        }

        code,
        pre {
            font-family: "Roboto Mono", monospace;
        }

        pre {
            white-space: pre-wrap;
        }

        h1,
        h2,
        h3 {
            color: var(--structure);
            font-family: "Roboto Condensed", sans-serif;
            font-weight: 400;
        }

        header {
            font-family: "Roboto Condensed", sans-serif;
            z-index: 5;
            color: black;
            margin-top: 4rem;
        }

        .header--title {
            font-weight: 400;
            font-size: 2rem;
            margin-bottom: 1rem;
        }

        .header--authors>span {
            color: var(--light);
            font-weight: 400;
        }

        main {
            padding-top: 8em !important;
            padding-bottom: 8em !important;
        }

        footer {
            background-color: var(--grey);
            font-weight: 300;
        }

        .abstract {
            color: var(--light);
            margin-bottom: 5em;
        }

        figure {
            max-width: 100%;
            border: 1px solid var(--pale);
            border-radius: 2px;
            margin: 1em 0.5em;
        }

        figure>img {
            max-width: 100%;
            display: block;
            margin: 0em auto;
        }

        figure>figcaption {
            padding: var(--inner-space);
        }

        table {
            width: 100%;
            min-width: 100%;
            border-collapse: collapse;
        }

        table>caption {
            padding: var(--inner-space);
        }

        figcaption,
        caption {
            color: var(--light);
            font-weight: 300;
        }

        thead {
            color: var(--structure);
        }

        td,
        th {
            padding: 3px;
        }

        tr {
            border-top: 1px solid var(--structure);
            border-bottom: 1px solid var(--structure);
        }

        tbody>tr:nth-child(even) {
            background: var(--grey)
        }

        ;

        tbody {
            border-bottom: 1px solid var(--structure);
        }

        #refs {
            font-size: 90%;
            column-count: 2;
        }

        #refs>.csl-entry {
            padding-bottom: 0.2em;
            padding-top: 0.2em;
        }

        .hanging-indent>.csl-entry {
            padding-left: 15px;
            text-indent: -15px;
        }

        .artifacts {
            text-align: center;
            border-radius: 4px;
            background-color: var(--grey);
            padding: 1em;
            margin-top: 4em;
        }

        .artifacts a{
            background-color: var(--structure);
            text-decoration: none;
            margin: 0px 2em;
            color: white;
            text-transform: capitalize;
            border-radius: 4px;
            text-align: center;
            padding: 4px 4px;
            width: 7em;
            display: inline-block;
        }    </style>
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.10.0-rc.1/dist/katex.min.css"
    integrity="sha384-D+9gmBxUQogRLqvARvNLmA9hS2x//eK1FhVb9PiU86gmcrBrJAQT8okdJ4LMp2uv" crossorigin="anonymous" />
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.10.0-rc.1/dist/katex.min.js"
    integrity="sha384-483A6DwYfKeDa0Q52fJmxFXkcPCFfnXMoXblOkJ4JcA8zATN6Tm78UNL72AKk+0O"
    crossorigin="anonymous"></script>
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.10.0-rc.1/dist/contrib/auto-render.min.js"
    integrity="sha384-yACMu8JWxKzSp/C1YV86pzGiQ/l1YUfE8oPuahJQxzehAjEt2GiQuy/BIvl9KyeF"
    crossorigin="anonymous"></script>
    <script>
        document.addEventListener("DOMContentLoaded", function () {
            renderMathInElement(document.body, {
                delimiters: [
                { left: "$$", right: "$$", display: true },
                { left: "$", right: "$", display: false }
                ]
            });
        });
    </script>
</head>

<body>
    <header>
        <div class="container">
            <div class="header--title">Graph embedding and transfer
learning can help predict potential species interaction networks despite
data limitations</div>
            <div class="header--authors">
                <span class="header--author">
                <a href="https://orcid.org/0000-0001-6067-1349" target="_blank">
                    Tanya Strydom
                </a>
                <sup class="affiliation">1,2,‡</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0003-0193-5441" target="_blank">
                    Salomé Bouskila
                </a>
                <sup class="affiliation">1</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0001-9051-0597" target="_blank">
                    Francis Banville
                </a>
                <sup class="affiliation">1,2,3</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0003-4036-977X" target="_blank">
                    Ceres Barros
                </a>
                <sup class="affiliation">4</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0002-2151-6693" target="_blank">
                    Dominique Caron
                </a>
                <sup class="affiliation">2,5</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0003-0452-6993" target="_blank">
                    Maxwell J Farrell
                </a>
                <sup class="affiliation">6</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0002-9935-1366" target="_blank">
                    Marie-Josée Fortin
                </a>
                <sup class="affiliation">6</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0003-3220-6161" target="_blank">
                    Victoria Hemming
                </a>
                <sup class="affiliation">7</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0002-4104-9463" target="_blank">
                    Benjamin Mercier
                </a>
                <sup class="affiliation">2,3</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0002-6004-4027" target="_blank">
                    Laura J. Pollock
                </a>
                <sup class="affiliation">2,5</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0001-5785-8321" target="_blank">
                    Rogini Runghen
                </a>
                <sup class="affiliation">8</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0002-3454-0633" target="_blank">
                    Giulio V. Dalla Riva
                </a>
                <sup class="affiliation">9</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0002-0735-5184" target="_blank">
                    Timothée Poisot
                </a>
                <sup class="affiliation">1,2,‡</sup>
                </span>
            </div>
        </div>
    </header>
    
    <main class="container">
        <section class="abstract">
            
            <ol>
                <li>Metawebs (networks of potential interactions within
            a species pool) are a powerful abstraction to understand how
            large-scale species interaction networks are
            structured.</li>
                <li>Because metawebs are typically expressed at large
            spatial and taxonomic scales, assembling them is a tedious
            and costly process; predictive methods can help circumvent
            the limitations in data deficiencies, by providing a first
            approximation of metawebs.</li>
                <li>One way to improve our ability to predict metawebs
            is to maximize available information by using graph
            embeddings, as opposed to an exhaustive list of species
            interactions. Graph embedding is an emerging field in
            machine learning that holds great potential for ecological
            problems.</li>
                <li>Here, we outline how the challenges associated with
            inferring metawebs line-up with the advantages of graph
            embeddings; followed by a discussion as to how the choice of
            the species pool has consequences on the reconstructed
            network, specifically as to the role of human-made (or
            arbitrarily assigned) boundaries and how these my influence
            ecological hypotheses.</li>
            </ol>

            <div class="artifacts">
                <a href="Strydom_et_al_2023_graph_embedding_draft.pdf" title="Draft PDF" target="_blank">Draft</a>
                <a href="Strydom_et_al_2023_graph_embedding_preprint.pdf" title="Preprint PDF" target="_blank">Preprint</a>
                <a href="Strydom_et_al_2023_graph_embedding.odt" title="OpenDocument (Word)" target="_blank">Libre Office</a>
            </div>
        </section>
        
        <article>
            <p>The ability to infer <em>potential</em> interactions
            could serve as a significant breakthrough in our ability to
            conceptualize species interaction networks over large
            spatial scales <span class="citation"
            data-cites="Hortal2015Seven">(Hortal et al., 2015)</span>.
            Reliable inferences would not only boost our understanding
            of the structure of species interaction networks, but also
            increase the amount of information that can be used for
            biodiversity management. In a recent overview of the field
            of ecological network prediction, <span class="citation"
            data-cites="Strydom2021Roadmap">Strydom, Catchen, et al.
            (2021)</span> identified two challenges of interest to the
            prediction of interactions at large scales. First, there is
            a relative scarcity of relevant data in most places globally
            – restricting the inference of interactions to locations
            where least required (and leaves us unable to make inference
            in data scarce regions); second, accurate predictors are
            important for accurate predictions, and the lack of methods
            that can leverage a small amount of <em>accurate</em> data
            is a serious impediment to our predictive ability. In most
            places, our most reliable biodiversity knowledge is that of
            a species pool where a set of potentially interacting
            species in a given area could occur: through the analysis of
            databases like the Global Biodiversity Information Facility
            (GBIF) or the International Union for the Conservation of
            Nature (IUCN), it is possible to construct a list of species
            for a region of interest; however inferring the potential
            interactions between these species still remains a
            challenge.</p>
            <p>Following the definition of <span class="citation"
            data-cites="Dunne2006Network">Dunne (2006)</span>, a metaweb
            is the ecological network analogue to the species pool;
            specifically, it inventories all <em>potential</em>
            interactions between species for a spatially delimited area
            (and so captures the <span
            class="math inline"><em>γ</em></span> diversity of
            interactions). The metaweb itself is not a prediction of
            local networks at specific locations within the spatial area
            it covers: it will have a different structure, notably by
            having a larger connectance <span class="citation"
            data-cites="Wood2015Effects">(see <em>e.g.</em> Wood et al.,
            2015)</span> and complexity <span class="citation"
            data-cites="Galiana2022Ecological">(see <em>e.g.</em>
            Galiana et al., 2022)</span>, from any of these local
            networks. These local networks (which capture the <span
            class="math inline"><em>α</em></span> diversity of
            interactions) are a subset of the metaweb’s species and its
            realized interactions, and have been called “metaweb
            realizations” <span class="citation"
            data-cites="Poisot2015Species">(Poisot et al., 2015)</span>.
            Differences between local networks and their metawebs are
            due to chance, species abundance and co-occurrence, local
            environmental conditions, and local distribution of
            functional traits, among others. Specifically, although
            co-occurrence can be driven by interactions <span
            class="citation" data-cites="Cazelles2016Theory">(Cazelles
            et al., 2016)</span>, co-occurrence alone is not a predictor
            of interactions <span class="citation"
            data-cites="Blanchet2020Cooccurrence Thurman2019Testing">(Blanchet
            et al., 2020; Thurman et al., 2019)</span>, and therefore
            the lack of co-occurrence cannot be used to infer the lack
            of a feasible interaction. Yet, recent results by <span
            class="citation" data-cites="Saravia2021Ecological">Saravia
            et al. (2021)</span> strongly suggested that local (metaweb)
            realizations only respond weakly to local conditions:
            instead, they reflect constraints inherited by the structure
            of their metaweb. This sets up the core goal of predictive
            network ecology as the prediction of metaweb structure, as
            it is required to accurately produce downscaled, local
            predictions.</p>
            <p>Because the metaweb represents the joint effect of
            functional, phylogenetic, and macroecological processes
            <span class="citation"
            data-cites="Morales-Castilla2015Inferring">(Morales-Castilla
            et al., 2015)</span>, it holds valuable ecological
            information. Specifically, it represents the “upper bounds”
            on what the composition of the local networks, given a local
            species pool, can be <span class="citation"
            data-cites="McLeod2021Sampling">(see <em>e.g.</em> McLeod et
            al., 2021)</span>; this information can help evaluate the
            ability of ecological assemblages to withstand the effects
            of, for example, climate change <span class="citation"
            data-cites="Fricke2022Effects">(Fricke et al., 2022)</span>.
            These local networks may be reconstructed given an
            appropriate knowledge of local species composition and
            provide information on the structure of food webs at finer
            spatial scales. This has been done for example for
            tree-galler-parasitoid systems <span class="citation"
            data-cites="Gravel2018Bringing">(Gravel et al.,
            2018)</span>, fish trophic interactions <span
            class="citation" data-cites="Albouy2019Marine">(Albouy et
            al., 2019)</span>, tetrapod trophic interactions <span
            class="citation"
            data-cites="Braga2019Spatial OConnor2020Unveiling">(J. Braga
            et al., 2019; O’Connor et al., 2020)</span>, and crop-pest
            networks <span class="citation"
            data-cites="Grunig2020Crop">(Grünig et al., 2020)</span>. In
            this contribution, we highlight the power of viewing (and
            constructing) metawebs as <em>probabilistic</em> objects in
            the context of low-probability interactions, discuss how a
            family of machine learning tools (graph embeddings and
            transfer learning) can be used to overcome data limitations
            to metaweb inference, and highlight how the use of metawebs
            introduces important questions for the field of network
            ecology.</p>
            <h1 id="a-metaweb-is-an-inherently-probabilistic-object">A
            metaweb is an inherently probabilistic object</h1>
            <p>Treating interactions as probabilistic (as opposed to
            binary) events is a more nuanced and realistic way to
            represent them. <span class="citation"
            data-cites="Dallas2017Predicting">Dallas et al.
            (2017)</span> suggested that most interactions (links) in
            ecological networks are cryptic, <em>i.e.</em> uncommon or
            hard to observe. This argument echoes <span class="citation"
            data-cites="Jordano2016Sampling">Jordano (2016)</span>:
            sampling ecological interactions is difficult because it
            requires first the joint observation of two species, and
            then the observation of their interaction. In addition, it
            is generally expected that weak or rare interactions will be
            more prevalent in networks than common or strong
            interactions <span class="citation"
            data-cites="Csermely2004Strong">(Csermely, 2004)</span>,
            compared to strong, persistent interactions; this is notably
            the case in food chains, wherein many weaker interactions
            are key to the stability of a system <span class="citation"
            data-cites="Neutel2002Stability">(Neutel et al.,
            2002)</span>. In the light of these observations, we expect
            to see an over-representation of low-probability (hereafter
            rare) interactions under a model that accurately predicts
            interaction probabilities. Although defining an interaction
            as ‘rare’ is perhaps not as straight forward as one may
            assume (rare in the context of likelihood of occurrence or
            the context of biologically plausible?) but for the context
            of the discussion in this manuscript the core idea is that
            construction of probabilistic networks affords us the nuance
            of capturing this inherit variability of interaction
            occurrence.</p>
            <p>Critically, the original metaweb definition, and indeed
            most past uses of metawebs, was based on the
            presence/absence of interactions. Moving towards
            <em>probabilistic</em> metawebs, by representing
            interactions as Bernoulli events <span class="citation"
            data-cites="Poisot2016Structure">(see <em>e.g.</em> Poisot
            et al., 2016)</span>, offers the opportunity to weigh these
            rare interactions appropriately. The inherent plasticity of
            interactions is important to capture: there have been
            documented instances of food webs undergoing rapid
            collapse/recovery cycles over short periods of time <span
            class="citation"
            data-cites="Pedersen2017Signatures">(<em>e.g.</em> Pedersen
            et al., 2017)</span>. Furthermore, because the structure of
            the metaweb cannot be known in advance, it is important to
            rely on predictive tools that do not assume a specific
            network topology for link prediction <span class="citation"
            data-cites="Gaucher2021Outlier">(Gaucher et al.,
            2021)</span>, but are able to work on generalizations of the
            network. These considerations emphasize why metaweb
            predictions should focus on quantitative (preferentially
            probabilistic) predictions, and this should constrain the
            suite of models that are appropriate for prediction.</p>
            <p>It is important to recall that a metaweb is intended as a
            catalogue of all potential (feasible) interactions, which is
            then filtered for a given application <span class="citation"
            data-cites="Morales-Castilla2015Inferring">(Morales-Castilla
            et al., 2015)</span>. It is therefore important to separate
            the interactions that happen “almost surely” (repeated
            observational data), “almost never” (repeated lack of
            evidence <em>or</em> evidence that the link is forbidden
            through <em>e.g.</em> trait mis-match), and interactions
            with a probability that lays somewhere in between <span
            class="citation" data-cites="Catchen2023Missing">(Catchen et
            al., 2023)</span>. In a sense, that most ecological
            interactions are elusive can call for a slightly different
            approach to sampling: once the common interactions are
            documented, the effort required in documenting each rare
            interaction will increase exponentially <span
            class="citation" data-cites="Jordano2016Sampling">(Jordano,
            2016)</span>. Recent proposals in other fields relying on
            machine learning approaches emphasize the idea that
            algorithms meant to predict, through the assumption that
            they approximate the process generating the data, can also
            act as data generators <span class="citation"
            data-cites="Hoffmann2019Machine">(Hoffmann et al.,
            2019)</span>. High quality observational data can be used to
            infer core rules underpinning network structure, and be
            supplemented with synthetic data coming from predictive
            models trained on them, thereby increasing the volume of
            information available for analysis. Indeed, <span
            class="citation" data-cites="Strydom2021Roadmap">Strydom,
            Catchen, et al. (2021)</span> suggested that knowing the
            metaweb may render the prediction of local networks easier,
            because it fixes an “upper bound” on which interactions can
            exist. In this context, a probabilistic metaweb represents
            an aggregation of informative priors on the biological
            feasibility of interactions, which is usually hard to obtain
            yet has possibly the most potential to boost our predictive
            ability of local networks <span class="citation"
            data-cites="Bartomeus2013Understanding Bartomeus2016Common">(Bartomeus,
            2013; Bartomeus et al., 2016)</span>. This would represent a
            departure from simple rules expressed at the network scale
            <span class="citation"
            data-cites="Williams2000Simple">(<em>e.g.</em> Williams
            &amp; Martinez, 2000)</span> to a view of network prediction
            based on learning the rules that underpin interactions
            <em>and</em> their variability <span class="citation"
            data-cites="Gupta2022Simultaneously">(Gupta et al.,
            2022)</span>.</p>
            <figure>
            <img src="figures/conceptual_2.png" id="fig:embedding"
            alt="Figure 1: The embedding process (A) can help to identify links (interactions) that may have been missed within the original community (represented by the orange dashed arrows, B). Transfer learning (D) allows for the prediction links (interactions) even when novel species (C) are included alongside the original community. This is achieved by learning using other relevant predictors (e.g. traits) in conjunction with the known interactions to infer latent values (E). Ultimately this allows us to predict links (interactions) for species external from the original sample (blue dashed arrows) as well as missing within sample links (F). Within this context the predicted (and original) networks as well as the ecological predictors used (green boxes) are products that can be quantified through measurements in the field, whereas the embedded as well as imputed matrices (purple box) are representative of a decomposition of the interaction matrices onto the embedding space" />
            <figcaption aria-hidden="true">Figure 1: The embedding
            process (<strong>A</strong>) can help to identify links
            (interactions) that may have been missed within the original
            community (represented by the orange dashed arrows,
            <strong>B</strong>). Transfer learning (<strong>D</strong>)
            allows for the prediction links (interactions) even when
            novel species (<strong>C</strong>) are included alongside
            the original community. This is achieved by learning using
            other relevant predictors (<em>e.g.</em> traits) in
            conjunction with the known interactions to infer latent
            values (<strong>E</strong>). Ultimately this allows us to
            predict links (interactions) for species external from the
            original sample (blue dashed arrows) as well as missing
            within sample links (<strong>F</strong>). Within this
            context the predicted (and original) networks as well as the
            ecological predictors used (green boxes) are products that
            can be quantified through measurements in the field, whereas
            the embedded as well as imputed matrices (purple box) are
            representative of a decomposition of the interaction
            matrices onto the embedding space</figcaption>
            </figure>
            <h1
            id="graph-embedding-offers-promises-for-the-inference-of-potential-interactions">Graph
            embedding offers promises for the inference of potential
            interactions</h1>
            <p>Graph (or network) embedding (fig. 1) is a family of
            machine learning techniques, whose main task is to learn a
            mapping function from a discrete graph to a continuous
            domain <span class="citation"
            data-cites="Arsov2019Network Chami2022Machine">(Arsov &amp;
            Mirceva, 2019; Chami et al., 2022)</span>. Their main goal
            is to learn a low dimensional vector representations of the
            graph (embeddings), such that its key properties
            (<em>e.g.</em> local or global structures) are retained in
            the embedding space <span class="citation"
            data-cites="Yan2005Graph">(Yan et al., 2005)</span>. The
            embedding space may, but will not necessarily, have lower
            dimensionality than the graph. Ecological networks are
            promising candidates for the routine application of
            embeddings, as they tend to possess a shared structural
            backbone <span class="citation"
            data-cites="BramonMora2018Identifying">(see <em>e.g.</em>
            Bramon Mora et al., 2018)</span>, which hints at structural
            invariants in empirical data. Assuming that these structural
            invariants are common enough, they would dominate the
            structure of networks, and therefore be adequately captured
            by the first (lower) dimensions of an embedding, without the
            need to measure derived aspects of their structure
            (<em>e.g.</em> motifs, paths, modularity, …).</p>
            <h2
            id="graph-embedding-produces-latent-variables-but-not-traits">Graph
            embedding produces latent variables (but not traits)</h2>
            <p>Before moving further, it is important to clarify the
            epistemic status of node values derived from embeddings:
            specifically, they are <em>not</em> functional traits, and
            therefore should not be interpreted in terms of effects or
            responses. As per the framework of <span class="citation"
            data-cites="Malaterre2019Functional">Malaterre et al.
            (2019)</span>, these values neither derive from, nor result
            in, changes in organismal performance, and should therefore
            not be used to quantify <em>e.g.</em> functional diversity.
            This holds true even when there are correlations between
            latent values and functional traits: although these enable
            an ecological discussion of how traits condition the
            structure of the network, the existence of a statistical
            relationship does not elevate the latent values to the
            status of functional traits.</p>
            <p>Rather than directly predicting biological rules <span
            class="citation" data-cites="Pichler2020Machine">(see
            <em>e.g.</em> Pichler et al., 2020 for an overview)</span>,
            which may be confounded by the sparse nature of graph data,
            learning embeddings works in the low-dimensional space that
            maximizes information about the network structure. This
            approach is further justified by the observation, for
            example, that the macro-evolutionary history of a network is
            adequately represented by some graph embeddings [Random dot
            product graphs (RDPG); see <span class="citation"
            data-cites="DallaRiva2016Exploring">Dalla Riva &amp;
            Stouffer (2016)</span>]. In a recent publication, <span
            class="citation" data-cites="Strydom2022Food">Strydom et al.
            (2022)</span> have used an embedding (based on RDPG) to
            project a metaweb of trophic interactions between European
            mammals, and transferred this information to mammals of
            Canada, using the phylogenetic distance between related
            clades to infer the values in the latent subspace into which
            the European metaweb was projected. By performing the RDPG
            step on re-constructed values, this approach yields a
            probabilistic trophic metaweb for mammals of Canada based on
            knowledge of European species, despite a limited (<span
            class="math inline">≈</span> 5%) taxonomic overlap, and
            illustrates how the values derived from an embedding can be
            used for prediction without being “traits” of the species
            they represent.</p>
            <h2
            id="ecological-networks-are-good-candidates-for-embedding">Ecological
            networks are good candidates for embedding</h2>
            <p>Food webs are inherently low-dimensional objects, and can
            be adequately represented with less than ten dimensions
            <span class="citation"
            data-cites="Braga2021Phylogenetic Eklof2013Dimensionality Braga2019Spatial">(J.
            Braga et al., 2019; M. P. Braga et al., 2021; Eklöf et al.,
            2013)</span>. Simulation results by <span class="citation"
            data-cites="Botella2022Appraisal">Botella et al.
            (2022)</span> suggested that there is no dominant method to
            identify architectural similarities between networks:
            multiple approaches need to be tested and compared to the
            network descriptor of interest on a problem-specific basis.
            This matches previous results on graph embedding, wherein
            different embedding algorithms yield different network
            embeddings <span class="citation"
            data-cites="Goyal2018Graph">(Goyal &amp; Ferrara,
            2018)</span>, calling for a careful selection of the
            problem-specific approach to use. In tbl. 1, we present a
            selection of common graph and node embedding methods,
            alongside examples of their use to predict interactions or
            statistical associations between species. These methods rely
            largely on linear algebra or pseudo-random walks on graphs.
            All forms of embeddings presented in tbl. 1 share the common
            property of summarizing their objects into (sets of) dense
            feature vectors, that capture the overall network structure,
            pairwise information on nodes, and emergent aspects of the
            network, in a compressed way (<em>i.e.</em> with some
            information loss, as we later discuss in the illustration).
            Node embeddings tend to focus on maintaining pairwise
            relationships (<em>i.e.</em> species interactions), while
            graph embeddings focus on maintaining the network structure
            (<em>i.e.</em> emergent properties). Nevertheless, some
            graph embedding techniques <span class="citation"
            data-cites="Wu2021Maximum">(like RDPG, see <em>e.g.</em> Wu
            et al., 2021)</span> will provide high-quality node-level
            embeddings while also preserving network structure.</p>
            <p>Graph embeddings <em>can</em> serve as a dimensionality
            reduction method. For example, RDPG <span class="citation"
            data-cites="Strydom2022Food">(Strydom et al., 2022)</span>
            and t-SVD [truncated Singular Value Decomposition; <span
            class="citation" data-cites="Poisot2021Imputing">Poisot et
            al. (2021)</span>] typically embed networks using fewer
            dimensions than the original network [the original network
            has as many dimensions as species, and as many informative
            dimensions as trophically unique species; <span
            class="citation" data-cites="Strydom2021Svd">Strydom, Dalla
            Riva, et al. (2021)</span>]. However, this is not
            necessarily the case – indeed, one may perform a PCA (a
            special case of SVD) to project the raw data into a subspace
            that improves the efficacy of t-SNE [t-distributed
            stochastic neighbor embedding; <span class="citation"
            data-cites="Maaten2009Learning">Maaten (2009)</span>]. There
            are many dimensionality reductions <span class="citation"
            data-cites="Anowar2021Conceptual">(Anowar et al.,
            2021)</span> that can be applied to an embedded network
            should the need for dimensionality reduction (for example
            for data visualization) arise. In brief, many graph
            embeddings <em>can</em> serve as dimensionality reduction
            steps, but not all do, neither do all dimensionality
            reduction methods provide adequate graph embedding
            capacities. In the next section (and fig. 1), we show how
            the amount of dimensionality reduction can affect the
            quality of the embedding.</p>
            <h2
            id="graph-embedding-has-been-under-used-in-the-prediction-of-species-interactions">Graph
            embedding has been under-used in the prediction of species
            interactions</h2>
            <p>One prominent family of approaches we do not discuss in
            the present manuscript is Graph Neural Networks [GNN; <span
            class="citation" data-cites="Zhou2020Graph">Zhou et al.
            (2020)</span>]. GNN are, in a sense, a method to embed a
            graph into a dense subspace, but belong to the family of
            deep learning methods, which has its own set of practices
            <span class="citation" data-cites="Goodfellow2016Deep">(see
            <em>e.g.</em> Goodfellow et al., 2016)</span>. An important
            issue with methods based on deep learning is that, because
            their parameter space is immense, the sample size of the
            data fed into them must be similarly large (typically
            thousands of instances). This is a requirement for the model
            to converge correctly during training, but this assumption
            is unlikely to be met given the size of datasets currently
            available for metawebs (or single time/location species
            interaction networks). This data volume requirement is
            mostly absent from the techniques we list below.
            Furthermore, GNN still have some challenges related to their
            shallow structure, and concerns related to scalability <span
            class="citation" data-cites="Gupta2021Graph">(see Gupta et
            al., 2021 for a review)</span>, which are mostly absent from
            the methods listed in tbl. 1. Assuming that the uptake of
            next-generation biomonitoring techniques does indeed deliver
            larger datasets on species interactions <span
            class="citation" data-cites="Bohan2017Nextgeneration">(Bohan
            et al., 2017)</span>, there is nevertheless the potential
            for GNN to become an applicable embedding/predictive
            technique in the coming years.</p>
            <div id="tbl:methods">
            <table>
            <caption>Table 1: Overview of some common graph embedding
            approaches, by type of embedded objects, alongside examples
            of their use in the prediction of species interactions.
            These methods have not yet been routinely used to predict
            species interactions; most examples that we identified were
            either statistical associations, or analogues to joint
            species distribution models. <span
            class="math inline"><sup><em>a</em></sup></span>:
            application is concerned with <em>statistical</em>
            interactions, which are not necessarilly direct biotic
            interactions; <span
            class="math inline"><sup><em>b</em></sup></span>:application
            is concerned with joint-SDM-like approach, which is also
            very close to statistical associations as opposed to direct
            biotic interactions. Given the need to evaluate different
            methods on a problem-specific basis, the fact that a lot of
            methods have not been used on network problems is an
            opportunity for benchmarking and method development. Note
            that the row for PCA also applies to kernel/probabilistic
            PCA, which are variations on the more general method of SVD.
            Note further that tSNE has been included because it is
            frequently used to embed graphs, including of species
            associations/interactions, despite not being strictly
            speaking, a graph embedding technique <span class="citation"
            data-cites="Chami2022Machine">(see <em>e.g.</em> Chami et
            al., 2022)</span>.</caption>
            <colgroup>
            <col style="width: 6%" />
            <col style="width: 7%" />
            <col style="width: 16%" />
            <col style="width: 12%" />
            <col style="width: 57%" />
            </colgroup>
            <thead>
            <tr class="header">
            <th>Method</th>
            <th>Object</th>
            <th>Technique</th>
            <th>Reference</th>
            <th>Application</th>
            </tr>
            </thead>
            <tbody>
            <tr class="odd">
            <td>tSNE</td>
            <td>nodes</td>
            <td>statistical divergence</td>
            <td><span class="citation"
            data-cites="Hinton2002Stochastic">Hinton &amp; Roweis
            (2002)</span></td>
            <td><span class="citation"
            data-cites="Cieslak2020Tdistributed">(Cieslak et al., 2020,
            species-environment responses <span
            class="math inline"><sup><em>a</em></sup></span>)</span>
            <span class="citation" data-cites="Gibb2021Data">(Gibb et
            al., 2021, host-virus network representation)</span></td>
            </tr>
            <tr class="even">
            <td>LINE</td>
            <td>nodes</td>
            <td>stochastic gradient descent</td>
            <td><span class="citation" data-cites="Tang2015Line">Tang et
            al. (2015)</span></td>
            <td></td>
            </tr>
            <tr class="odd">
            <td>SDNE</td>
            <td>nodes</td>
            <td>gradient descent</td>
            <td><span class="citation"
            data-cites="Wang2016Structural">D. Wang et al.
            (2016)</span></td>
            <td></td>
            </tr>
            <tr class="even">
            <td>node2vec</td>
            <td>nodes</td>
            <td>stochastic gradient descent</td>
            <td><span class="citation"
            data-cites="Grover2016Node2vec">Grover &amp; Leskovec
            (2016)</span></td>
            <td></td>
            </tr>
            <tr class="odd">
            <td>HARP</td>
            <td>nodes</td>
            <td>meta-strategy</td>
            <td><span class="citation" data-cites="Chen2017Harp">H. Chen
            et al. (2017)</span></td>
            <td></td>
            </tr>
            <tr class="even">
            <td>DMSE</td>
            <td>joint nodes</td>
            <td>deep neural network</td>
            <td><span class="citation" data-cites="Chen2017Deep">D. Chen
            et al. (2017)</span></td>
            <td><span class="citation" data-cites="Chen2017Deep">(D.
            Chen et al., 2017, species-environment interactions <span
            class="math inline"><sup><em>b</em></sup></span>)</span></td>
            </tr>
            <tr class="odd">
            <td>graph2vec</td>
            <td>sub-graph</td>
            <td>skipgram network</td>
            <td><span class="citation"
            data-cites="Narayanan2017Graph2vec">Narayanan et al.
            (2017)</span></td>
            <td></td>
            </tr>
            <tr class="even">
            <td>RDPG</td>
            <td>graph</td>
            <td>SVD</td>
            <td><span class="citation"
            data-cites="Young2007Random">Young &amp; Scheinerman
            (2007)</span></td>
            <td><span class="citation"
            data-cites="DallaRiva2016Exploring">(Dalla Riva &amp;
            Stouffer, 2016, trophic interactions)</span> <span
            class="citation" data-cites="Poisot2021Imputing">(Poisot et
            al., 2021, host-virus network prediction)</span></td>
            </tr>
            <tr class="odd">
            <td>GLEE</td>
            <td>graph</td>
            <td>Laplacian eigenmap</td>
            <td><span class="citation"
            data-cites="Torres2020Glee">Torres et al. (2020)</span></td>
            <td></td>
            </tr>
            <tr class="even">
            <td>DeepWalk</td>
            <td>graph</td>
            <td>stochastic gradient descent</td>
            <td><span class="citation"
            data-cites="Perozzi2014Deepwalk">Perozzi et al.
            (2014)</span></td>
            <td><span class="citation"
            data-cites="Wardeh2021Predicting">(Wardeh et al., 2021,
            host-virus interactions)</span></td>
            </tr>
            <tr class="odd">
            <td>GraphKKE</td>
            <td>graph</td>
            <td>stochastic differential equation</td>
            <td><span class="citation"
            data-cites="Melnyk2020Graphkke">Melnyk et al.
            (2020)</span></td>
            <td><span class="citation"
            data-cites="Melnyk2020Graphkke">(Melnyk et al., 2020,
            microbiome species associations <span
            class="math inline"><sup><em>a</em></sup></span>)</span></td>
            </tr>
            <tr class="even">
            <td>FastEmbed</td>
            <td>graph</td>
            <td>eigen decomposition</td>
            <td><span class="citation"
            data-cites="Ramasamy2015Compressive">Ramasamy &amp; Madhow
            (2015)</span></td>
            <td></td>
            </tr>
            <tr class="odd">
            <td>PCA</td>
            <td>graph</td>
            <td>eigen decomposition</td>
            <td><span class="citation"
            data-cites="Surendran2013Graph">Surendran (2013)</span></td>
            <td><span class="citation"
            data-cites="Strydom2021Roadmap">(Strydom, Catchen, et al.,
            2021, host-parasite interactions)</span></td>
            </tr>
            <tr class="even">
            <td>Joint methods</td>
            <td>multiple graphs</td>
            <td>multiple strategies</td>
            <td><span class="citation" data-cites="Wang2021Joint">S.
            Wang et al. (2021)</span></td>
            <td></td>
            </tr>
            </tbody>
            </table>
            </div>
            <p>The popularity of graph embedding techniques in machine
            learning is more than the search for structural invariants:
            graphs are discrete objects, and machine learning techniques
            tend to handle continuous data better. Bringing a sparse
            graph into a continuous, dense vector space <span
            class="citation" data-cites="Xu2021Understanding">(Xu,
            2021)</span> opens up a broader variety of predictive
            algorithms, notably of the sort that are able to predict
            events as probabilities <span class="citation"
            data-cites="Murphy2022Probabilistic">(Murphy, 2022)</span>.
            Furthermore, the projection of the graph itself is a
            representation that can be learned; <span class="citation"
            data-cites="Runghen2021Exploiting">Runghen et al.
            (2021)</span>, for example, used a neural network to learn
            the embedding of a network in which not all interactions
            were known, based on the nodes’ metadata. This example has
            many parallels in ecology (see fig. 1 <strong>C</strong>),
            in which node metadata can be represented by phylogeny,
            abundance, or functional traits. Using phylogeny as a source
            of information assumes (or strives to capture) the action of
            evolutionary processes on network structure, which at least
            for food webs have been well documented <span
            class="citation"
            data-cites="Braga2021Phylogenetic DallaRiva2016Exploring Eklof2016Phylogenetic Stouffer2007Evidence Stouffer2012Evolutionary">(M.
            P. Braga et al., 2021; Dalla Riva &amp; Stouffer, 2016;
            Eklöf &amp; Stouffer, 2016; Stouffer et al., 2012; Stouffer
            et al., 2007)</span>; similarly, the use of functional
            traits assumes that interactions can be inferred from the
            knowledge of trait-matching rules, which is similarly well
            supported in the empirical literature <span class="citation"
            data-cites="Bartomeus2013Understanding Bartomeus2016Common Goebel2023Body Gravel2013Inferring">(Bartomeus,
            2013; Bartomeus et al., 2016; Goebel et al., 2023; Gravel et
            al., 2013)</span>. Relating this information to an embedding
            rather than a list of network measures would allow to
            capture their effect on the more fundamental aspects of
            network structure; conversely, the absence of a phylogenetic
            or functional signal may suggest that evolutionary/trait
            processes are not strong drivers of network structure,
            therefore opening a new way to perform hypothesis
            testing.</p>
            <h1 id="an-illustration-of-metaweb-embedding">An
            illustration of metaweb embedding</h1>
            <p>In this section, we illustrate the embedding of a
            collection of bipartite networks collected by <span
            class="citation" data-cites="Hadfield2014Tale">Hadfield et
            al. (2014)</span>, using t-SVD and RDPG. Briefly, an RDPG
            decomposes a network into two subspaces (left and right),
            which are matrices that when multiplied give an
            approximation of the original network. RDPG has the
            particularly desirable properties of being a graph embedding
            technique that produces relevant node-level feature vectors,
            and provides good approximations of graphs with varied
            structures <span class="citation"
            data-cites="Athreya2017Statistical">(Athreya et al.,
            2017)</span>. The code to reproduce this example is
            available as supplementary material <span class="citation"
            data-cites="Strydom2021Roadmap">(note, for the sake of
            comparison, that Strydom, Catchen, et al., 2021 have an
            example using embedding through PCA followed by prediction
            using a deep neural network on the same dataset)</span>. The
            resulting (binary) metaweb <span
            class="math inline">ℳ</span> has 2131 interactions between
            206 parasites and 121 hosts, and its adjacency matrix has
            full rank (<em>i.e.</em> it represents a space with 121
            dimensions). All analyses were done using Julia <span
            class="citation" data-cites="Bezanson2017Julia">(Bezanson et
            al., 2017)</span> version 1.7.2, <em>Makie.jl</em> <span
            class="citation" data-cites="Danisch2021Makie">(Danisch
            &amp; Krumbiegel, 2021)</span>, and
            <em>EcologicalNetworks.jl</em> <span class="citation"
            data-cites="Poisot2019Ecologicalnetworks">(Poisot et al.,
            2019)</span>.</p>
            <figure>
            <img src="figures/illustration-part1.png"
            id="fig:illustration1"
            alt="Figure 2: Validation of an embedding for a host-parasite metaweb, using Random Dot Product Graphs. A, decrease in approximation error as the number of dimensions in the subspaces increases. B, increase in cumulative variance explained as the number of ranks considered increases; in A and B, the dot represents the point of inflexion in the curve (at rank 39) estimated using the finite differences method. C, position of hosts and parasites in the space of latent variables on the first and second dimensions of their respective subspaces (the results have been clamped to the unit interval). D, predicted interaction weight from the RDPG based on the status of the species pair in the metaweb." />
            <figcaption aria-hidden="true">Figure 2: Validation of an
            embedding for a host-parasite metaweb, using Random Dot
            Product Graphs. <strong>A</strong>, decrease in
            approximation error as the number of dimensions in the
            subspaces increases. <strong>B</strong>, increase in
            cumulative variance explained as the number of ranks
            considered increases; in <strong>A</strong> and
            <strong>B</strong>, the dot represents the point of
            inflexion in the curve (at rank 39) estimated using the
            finite differences method. <strong>C</strong>, position of
            hosts and parasites in the space of latent variables on the
            first and second dimensions of their respective subspaces
            (the results have been clamped to the unit interval).
            <strong>D</strong>, predicted interaction weight from the
            RDPG based on the status of the species pair in the
            metaweb.</figcaption>
            </figure>
            <p>In fig. 2, we focus on some statistical checks of the
            embedding. In panel <strong>A</strong>, we show that the
            averaged <span
            class="math inline"><em>L</em><sub>2</sub></span> loss
            (<em>i.e.</em> the sum of squared errors) between the
            empirical and reconstructed metaweb decreases when the
            number of dimensions (rank) of the subspace increases, with
            an inflection at 39 dimensions (out of 120 initially)
            according to the finite differences method. As discussed by
            <span class="citation"
            data-cites="Runghen2021Exploiting">Runghen et al.
            (2021)</span>, there is often a trade-off between the number
            of dimensions to use (more dimensions are more
            computationally demanding) and the quality of the
            representation. In panel <strong>B</strong>, we show the
            increase in cumulative variance explained at each rank, and
            visualize that using 39 ranks explains about 70% of the
            variance in the empirical metaweb. This is a different
            information from the <span
            class="math inline"><em>L</em><sub>2</sub></span> loss
            (which is averaged across interactions), as it works on the
            eigenvalues of the embedding, and therefore captures
            higher-level features of the network. In panel
            <strong>C</strong>, we show positions of hosts and parasites
            on the first two dimensions of the left and right subspaces.
            Note that these values largely skew negative, because the
            first dimensions capture the coarse structure of the
            network: most pairs of species do not interact, and
            therefore have negative values. Finally in panel
            <strong>D</strong>, we show the predicted weight
            (<em>i.e.</em> the result of the multiplication of the RDGP
            subspaces at a rank of 39) as a function of whether the
            interactions are observed, not-observed, or unknown due to
            lack of co-occurrence in the original dataset. This reveals
            that the observed interactions have higher predicted
            weights, although there is some overlap; the usual approach
            to identify potential interactions based on this information
            would be a thresholding analysis, which is outside the scope
            of this manuscript (and is done in the papers cited in this
            illustration). Because the values returned from RDPG are not
            bound to the unit interval, we performed a clamping of the
            weights to the unit space, showing a one-inflation in
            documented interactions, and a zero-inflation in other
            species pairs. This last figure crosses from the statistical
            into the ecological, by showing that species pairs with no
            documented co-occurrence have weights that are not
            distinguishable from species pairs with no documented
            interactions, suggesting that (as befits a host-parasite
            model) the ability to interact is a strong predictor of
            co-occurrence.</p>
            <figure>
            <img src="figures/illustration-part2.png"
            id="fig:illustration2"
            alt="Figure 3: Ecological analysis of an embedding for a host-parasite metaweb, using Random Dot Product Graphs. A, relationship between the number of parasites and position along the first axis of the right-subspace for all hosts, showing that the embedding captures elements of network structure at the species scale. B, weak relationship between the body mass of hosts (in grams) and the position alongside the same dimension. C, weak relationship between body mass of hosts and parasite richness. D, distribution of positions alongside the same axis for hosts grouped by taxonomic family." />
            <figcaption aria-hidden="true">Figure 3: Ecological analysis
            of an embedding for a host-parasite metaweb, using Random
            Dot Product Graphs. <strong>A</strong>, relationship between
            the number of parasites and position along the first axis of
            the right-subspace for all hosts, showing that the embedding
            captures elements of network structure at the species scale.
            <strong>B</strong>, weak relationship between the body mass
            of hosts (in grams) and the position alongside the same
            dimension. <strong>C</strong>, weak relationship between
            body mass of hosts and parasite richness.
            <strong>D</strong>, distribution of positions alongside the
            same axis for hosts grouped by taxonomic
            family.</figcaption>
            </figure>
            <p>The results of fig. 2 show that we can extract an
            embedding of the metaweb that captures enough variance to be
            relevant; specifically, this is true for both <span
            class="math inline"><em>L</em><sub>2</sub></span> loss
            (indicating that RDPG is able to capture pairwise processes)
            and the cumulative variance explained (indicating that RDPG
            is able to capture network-level structure). Therefore, in
            fig. 3, we relate the values of latent variables for hosts
            to different ecologically-relevant data. In panel
            <strong>A</strong>, we show that host with a higher value on
            the first dimension have fewer parasites. This relates to
            the body size of hosts in the <em>PanTHERIA</em> database
            <span class="citation"
            data-cites="Jones2009Pantheria">(Jones et al., 2009)</span>,
            as shown in panel <strong>B</strong>: interestingly, the
            position on the first axis is only weakly correlated to body
            mass of the host; this matches well established results
            showing that body size/mass is not always a direct predictor
            of parasite richness in terrestrial mammals <span
            class="citation" data-cites="Morand1998Density">(Morand
            &amp; Poulin, 1998)</span>, a result we observe in panel
            <strong>C</strong>. Finally, in panel <strong>D</strong>, we
            can see how different taxonomic families occupy different
            positions on the first axis, with <em>e.g.</em> Sciuridae
            being biased towards higher values. These results show how
            we can look for ecological informations in the output of
            network embeddings, which can further be refined into the
            selection of predictors for transfer learning.</p>
            <h1
            id="the-metaweb-merges-ecological-hypotheses-and-practices">The
            metaweb merges ecological hypotheses and practices</h1>
            <p>Metaweb inference seeks to provide information about the
            interactions between species at a large spatial scale,
            typically a scale large enough to be considered of
            biogeographic relevance (indeed, many of the examples
            covered in the introduction span areas larger than a
            country, some of them global). But as <span class="citation"
            data-cites="Herbert1965Dune">Herbert (1965)</span>
            rightfully pointed out, “[y]ou can’t draw neat lines around
            planet-wide problems”; any inference of a metaweb must
            therefore contend with several novel, interwoven, families
            of problems. In this section, we outline three that we think
            are particularly important, and can discuss how they may
            addressed with subsequent data analysis or simulations, and
            how they emerge in the specific context of using embeddings;
            some of these issues are related to the application of these
            methods at the science-policy interface.</p>
            <h2
            id="identifying-the-properties-of-the-network-to-embed">Identifying
            the properties of the network to embed</h2>
            <p>If the initial metaweb is too narrow in scope, notably
            from a taxonomic point of view, the chances of finding
            another area with enough related species (through
            phylogenetic relatedness or similarity of functional traits)
            to make a reliable inference decreases. This is because
            transfer requires similarity (fig. 1). A diagnostic for the
            lack of similar species would likely be large confidence
            intervals during estimation of the values in the low-rank
            space. In other words, the representation of the original
            graph is difficult to transfer to the new problem.
            Alternatively, if the initial metaweb is too large
            (taxonomically), then the resulting embeddings would need to
            represent interactions between taxonomic groups that are not
            present in the new location. This would lead to a much
            higher variance in the starting dataset, and to
            under-dispersion in the target dataset, resulting in the
            potential under or over estimation of the strength of new
            predicted interactions. <span class="citation"
            data-cites="Llewelyn2022Predicting">Llewelyn et al.
            (2022)</span> provided compelling evidence for these
            situations by showing that, even at small spatial scales,
            the transfer of information about interactions becomes more
            challenging when areas rich with endemic species are
            considered. The lack of well documented metawebs is
            currently preventing the development of more concrete
            guidelines. The question of phylogenetic relatedness and
            distribution is notably relevant if the metaweb is assembled
            in an area with mostly endemic species (<em>e.g.</em> a
            system that has undergone recent radiation or that has
            remained in isolation for a long period of time might not
            have an analogous system with which to draw knowledge from),
            and as with every predictive algorithm, there is room for
            the application of our best ecological judgement. Because
            this problem relates to distribution of species in the
            geographic or phylogenetic space, it can certainly be
            approached through assessing the performance of embedding
            transfer in simulated starting/target species pools.</p>
            <h2
            id="identifying-the-scope-of-the-prediction-to-perform">Identifying
            the scope of the prediction to perform</h2>
            <p>The area for which we seek to predict the metaweb should
            determine the species pool on which the embedding is
            performed. Metawebs can be constructed by assigning
            interactions in a list of species within specific regions.
            The upside of this approach is that information relevant for
            the construction of this dataset is likely to exist, as
            countries usually set conservation goals at the national
            level <span class="citation"
            data-cites="Buxton2021Key">(Buxton et al., 2021)</span>, and
            as quantitative instruments are consequently designed to
            work at these scales <span class="citation"
            data-cites="Turak2017Using">(Turak et al., 2017)</span>;
            specific strategies are often enacted at smaller scales,
            nested within a specific country <span class="citation"
            data-cites="Ray2021Biodiversity">(Ray et al., 2021)</span>.
            However, there is no guarantee that these arbitrary
            boundaries are meaningful. In fact, we do not have a
            satisfying answer to the question of “where does an
            ecological network stop?”, the answer to which would dictate
            the spatial span to embed/predict. Recent results by <span
            class="citation" data-cites="Martins2022Global">Martins et
            al. (2022)</span> suggested that networks are shaped within
            eco-regions, with abrupt structural transitions from an
            eco-region to the next. Should this trend hold generally,
            this would provide an ecologically-relevant scale at which
            metawebs can be downscaled and predicted. Other solutions
            could leverage network-area relationships to identify areas
            in which networks are structurally similar <span
            class="citation"
            data-cites="Fortin2021Network Galiana2018Spatial Galiana2022Ecological">(see
            <em>e.g.</em> Fortin et al., 2021; Galiana et al., 2022,
            2018)</span>. Both of these solutions require ample
            pre-existing information about the network in space.
            Nevertheless, the inclusion of species for which we have
            data but that are not in the right spatial extent
            <em>may</em> improve the performance of approaches based on
            embedding and transfer, <em>if</em> they increase the
            similarity between the target and destination network. This
            proposal can specifically be evaluated by adding nodes to
            the network to embed, and assessing the performance of
            predictive models <span class="citation"
            data-cites="Llewelyn2022Predicting">(see <em>e.g.</em>
            Llewelyn et al., 2022)</span>.</p>
            <h2
            id="minding-legacies-shaping-ecological-datasets">Minding
            legacies shaping ecological datasets</h2>
            <p>In large parts of the world, boundaries that delineate
            geographic regions are merely a reflection the legacy of
            settler colonialism, which drives global disparity in
            capacity to collect and publish ecological data. Applying
            any embedding to biased data does not debias them, but
            rather embeds these biases, propagating them to the models
            using embeddings to make predictions. Furthermore, the use
            of ecological data itself is not an apolitical act <span
            class="citation" data-cites="Nost2021Political">(Nost &amp;
            Goldstein, 2021)</span>: data infrastructures tend to be
            designed to answer questions within national boundaries
            (therefore placing contingencies on what is available to be
            embedded), their use often drawing upon, and reinforcing,
            territorial statecraft <span class="citation"
            data-cites="Barrett2005Environment">(see <em>e.g.</em>
            Barrett, 2005)</span>. As per <span class="citation"
            data-cites="Machen2021Thinking">Machen &amp; Nost
            (2021)</span>, these biases are particularly important to
            consider when knowledge generated algorithmically is used to
            supplement or replace human decision-making, especially for
            governance (<em>e.g.</em> enacting conservation decisions on
            the basis of model prediction). As information on networks
            is increasingly leveraged for conservation actions <span
            class="citation"
            data-cites="Eero2021Use Naman2022Food Stier2017Integrating">(see
            <em>e.g.</em> Eero et al., 2021; Naman et al., 2022; Stier
            et al., 2017)</span>, the need to appraise and correct
            biases that are unwittingly propagated to algorithms when
            embedded from the original data is immense. These
            considerations are even more urgent in the specific context
            of biodiversity data. Long-term colonial legacies still
            shape taxonomic composition to this day <span
            class="citation"
            data-cites="Lenzner2022Naturalized Raja2022Colonialism">(Lenzner
            et al., 2022; Raja, 2022)</span>, and much shorter-term
            changes in taxonomic and genetic richness of wildlife
            emerged through environmental racism <span class="citation"
            data-cites="Schmidt2022Systemic">(Schmidt &amp; Garroway,
            2022)</span>. Thus, the set of species found at a specific
            location is not only as the result of a response to
            ecological processes separate from human influence, but also
            the result of human-environment interaction as well as the
            result legislative/political histories.</p>
            <h1
            id="conclusion-metawebs-predictions-and-people">Conclusion:
            metawebs, predictions, and people</h1>
            <p>Predictive approaches in ecology, regardless of the scale
            at which they are deployed and the intent of their
            deployment, originate in the framework that contributed to
            the ongoing biodiversity crisis <span class="citation"
            data-cites="Adam2014Elephant">(Adam, 2014)</span> and
            reinforced environmental injustice <span class="citation"
            data-cites="Choudry2013Saving Dominguez2020Decolonising">(Choudry,
            2013; Domínguez &amp; Luoma, 2020)</span>. The risk of
            embedding this legacy in our models is real, especially when
            the impact of this legacy on species pools is being
            increasingly documented. This problem can be addressed by
            re-framing the way we interact with models, especially when
            models are intended to support conservation actions.
            Particularly on territories that were traditionally
            stewarded by Indigenous people, we must interrogate how
            predictive approaches and the biases that underpin them can
            be put to task in accompanying Indigenous principles of land
            management <span class="citation"
            data-cites="Eichhorn2019Steps Nokmaq2021Awakening">(Eichhorn
            et al., 2019; No’kmaq et al., 2021)</span>. The discussion
            of “algorithm-in-the-loop” approaches that is now pervasive
            in the machine learning community provides examples of why
            this is important. Human-algorithm interactions are
            notoriously difficult and can yield adverse effects <span
            class="citation"
            data-cites="Green2019Disparate Stevenson2021Algorithmic">(Green
            &amp; Chen, 2019; Stevenson &amp; Doleac, 2021)</span>,
            suggesting the need to systematically study them for the
            specific purpose of, here, biodiversity governance.
            Improving the algorithmic literacy of decision makers is
            part of the solution <span class="citation"
            data-cites="Lamba2019Deep MoseboFernandes2020Machine">(<em>e.g.</em>
            Lamba et al., 2019; Mosebo Fernandes et al., 2020)</span>,
            as we can reasonably expect that model outputs will be
            increasingly used to drive policy decisions <span
            class="citation"
            data-cites="Weiskopf2022Increasing">(Weiskopf et al.,
            2022)</span>. Our discussion of these approaches need to go
            beyond the technical and statistical, and into the
            governance consequences they can have. To embed data also
            embeds historical and contemporary biases that acted on
            these data, both because they shaped the ecological
            processes generating them, and the global processes leading
            to their measurement and publication. For a domain as vast
            as species interaction networks, these biases exist at
            multiple scales along the way, and a challenge for
            prediction is not only to develop (or adopt) new
            quantitative tools, but to assess the behavior of these
            tools in the proper context.</p>
            <p><strong>Acknowledgements:</strong> We acknowledge that
            this study was conducted on land within the traditional
            unceded territory of the Saint Lawrence Iroquoian,
            Anishinabewaki, Mohawk, Huron-Wendat, and Omàmiwininiwak
            nations. TP, TS, DC, and LP received funding from the
            Canadian Institute for Ecology &amp; Evolution. FB is funded
            by the Institute for Data Valorization (IVADO). TS, SB, and
            TP are funded by a donation from the Courtois Foundation. CB
            was awarded a Mitacs Elevate Fellowship no. IT12391, in
            partnership with fRI Research, and also acknowledges funding
            from Alberta Innovates and the Forest Resources Improvement
            Association of Alberta. M-JF acknowledges funding from NSERC
            Discovery Grant and NSERC CRC. RR is funded by New Zealand’s
            Biological Heritage Ngā Koiora Tuku Iho National Science
            Challenge, administered by New Zealand Ministry of Business,
            Innovation, and Employment. BM is funded by the NSERC
            Alexander Graham Bell Canada Graduate Scholarship and the
            FRQNT master’s scholarship. LP acknowledges funding from
            NSERC Discovery Grant (NSERC RGPIN-2019-05771). TP
            acknowledges financial support from the Fondation Courtois,
            and NSERC through the Discovery Grants and Discovery
            Accelerator Supplement programs. MJF is supported by an
            NSERC PDF and an RBC Post-Doctoral Fellowship.</p>
            <p><strong>Conflict of interest:</strong> The authors have
            no conflict of interests to disclose</p>
            <p><strong>Authors’ contributions:</strong> TS, and TP
            conceived the ideas discussed in the manuscript. All authors
            contributed to writing and editing the manuscript.</p>
            <p><strong>Data availability:</strong> There is no data
            associated with this manuscript.</p>
            <h1 class="unnumbered" id="references">References</h1>
            <div id="refs"
            class="references csl-bib-body hanging-indent"
            data-line-spacing="2" role="doc-bibliography">
            <div id="ref-Adam2014Elephant" class="csl-entry"
            role="doc-biblioentry">
            Adam, R. (2014). <em>Elephant treaties: The Colonial legacy
            of the biodiversity crisis</em>. UPNE.
            </div>
            <div id="ref-Albouy2019Marine" class="csl-entry"
            role="doc-biblioentry">
            Albouy, C., Archambault, P., Appeltans, W., Araújo, M. B.,
            Beauchesne, D., Cazelles, K., Cirtwill, A. R., Fortin,
            M.-J., Galiana, N., Leroux, S. J., Pellissier, L., Poisot,
            T., Stouffer, D. B., Wood, S. A., &amp; Gravel, D. (2019).
            The marine fish food web is globally connected. <em>Nature
            Ecology &amp; Evolution</em>, <em>3</em>(8, 8), 1153–1161.
            <a
            href="https://doi.org/10.1038/s41559-019-0950-y">https://doi.org/10.1038/s41559-019-0950-y</a>
            </div>
            <div id="ref-Anowar2021Conceptual" class="csl-entry"
            role="doc-biblioentry">
            Anowar, F., Sadaoui, S., &amp; Selim, B. (2021). Conceptual
            and empirical comparison of dimensionality reduction
            algorithms (PCA, KPCA, LDA, MDS, SVD, LLE, ISOMAP, LE, ICA,
            t-SNE). <em>Computer Science Review</em>, <em>40</em>,
            100378. <a
            href="https://doi.org/10.1016/j.cosrev.2021.100378">https://doi.org/10.1016/j.cosrev.2021.100378</a>
            </div>
            <div id="ref-Arsov2019Network" class="csl-entry"
            role="doc-biblioentry">
            Arsov, N., &amp; Mirceva, G. (2019). <em>Network Embedding:
            An Overview</em>. <a
            href="http://arxiv.org/abs/1911.11726">http://arxiv.org/abs/1911.11726</a>
            </div>
            <div id="ref-Athreya2017Statistical" class="csl-entry"
            role="doc-biblioentry">
            Athreya, A., Fishkind, D. E., Levin, K., Lyzinski, V., Park,
            Y., Qin, Y., Sussman, D. L., Tang, M., Vogelstein, J. T.,
            &amp; Priebe, C. E. (2017). <em>Statistical inference on
            random dot product graphs: A survey</em> (No.
            arXiv:1709.05454). arXiv. <a
            href="http://arxiv.org/abs/1709.05454">http://arxiv.org/abs/1709.05454</a>
            </div>
            <div id="ref-Barrett2005Environment" class="csl-entry"
            role="doc-biblioentry">
            Barrett, S. (2005). <em>Environment and Statecraft: The
            Strategy of Environmental Treaty-Making</em> (1st ed.).
            Oxford University PressOxford. <a
            href="https://doi.org/10.1093/0199286094.001.0001">https://doi.org/10.1093/0199286094.001.0001</a>
            </div>
            <div id="ref-Bartomeus2013Understanding" class="csl-entry"
            role="doc-biblioentry">
            Bartomeus, I. (2013). Understanding linkage rules in
            plant-pollinator networks by using hierarchical models that
            incorporate pollinator detectability and plant traits.
            <em>PloS One</em>, <em>8</em>(7), e69200. <a
            href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0069200">http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0069200</a>
            </div>
            <div id="ref-Bartomeus2016Common" class="csl-entry"
            role="doc-biblioentry">
            Bartomeus, I., Gravel, D., Tylianakis, J. M., Aizen, M. A.,
            Dickie, I. A., &amp; Bernard-Verdier, M. (2016). A common
            framework for identifying linkage rules across different
            types of interactions. <em>Functional Ecology</em>,
            <em>30</em>(12), 1894–1903. <a
            href="http://onlinelibrary.wiley.com/doi/10.1111/1365-2435.12666/full">http://onlinelibrary.wiley.com/doi/10.1111/1365-2435.12666/full</a>
            </div>
            <div id="ref-Bezanson2017Julia" class="csl-entry"
            role="doc-biblioentry">
            Bezanson, J., Edelman, A., Karpinski, S., &amp; Shah, V. B.
            (2017). Julia: A Fresh Approach to Numerical Computing.
            <em>SIAM Review</em>, <em>59</em>(1), 65–98. <a
            href="https://doi.org/10.1137/141000671">https://doi.org/10.1137/141000671</a>
            </div>
            <div id="ref-Blanchet2020Cooccurrence" class="csl-entry"
            role="doc-biblioentry">
            Blanchet, F. G., Cazelles, K., &amp; Gravel, D. (2020).
            Co-occurrence is not evidence of ecological interactions.
            <em>Ecology Letters</em>.
            </div>
            <div id="ref-Bohan2017Nextgeneration" class="csl-entry"
            role="doc-biblioentry">
            Bohan, D. A., Vacher, C., Tamaddoni-Nezhad, A., Raybould,
            A., Dumbrell, A. J., &amp; Woodward, G. (2017).
            Next-Generation Global Biomonitoring: Large-scale, Automated
            Reconstruction of Ecological Networks. <em>Trends in Ecology
            &amp; Evolution</em>. <a
            href="https://doi.org/10.1016/j.tree.2017.03.001">https://doi.org/10.1016/j.tree.2017.03.001</a>
            </div>
            <div id="ref-Botella2022Appraisal" class="csl-entry"
            role="doc-biblioentry">
            Botella, C., Dray, S., Matias, C., Miele, V., &amp;
            Thuiller, W. (2022). An appraisal of graph embeddings for
            comparing trophic network architectures. <em>Methods in
            Ecology and Evolution</em>, <em>13</em>(1), 203–216. <a
            href="https://doi.org/10.1111/2041-210X.13738">https://doi.org/10.1111/2041-210X.13738</a>
            </div>
            <div id="ref-Braga2019Spatial" class="csl-entry"
            role="doc-biblioentry">
            Braga, J., Pollock, L. J., Barros, C., Galiana, N., Montoya,
            J. M., Gravel, D., Maiorano, L., Montemaggiori, A.,
            Ficetola, G. F., Dray, S., &amp; Thuiller, W. (2019).
            Spatial analyses of multi-trophic terrestrial vertebrate
            assemblages in Europe. <em>Global Ecology and
            Biogeography</em>, <em>28</em>(11), 1636–1648. <a
            href="https://doi.org/10.1111/geb.12981">https://doi.org/10.1111/geb.12981</a>
            </div>
            <div id="ref-Braga2021Phylogenetic" class="csl-entry"
            role="doc-biblioentry">
            Braga, M. P., Janz, N., Nylin, S., Ronquist, F., &amp;
            Landis, M. J. (2021). Phylogenetic reconstruction of
            ancestral ecological networks through time for pierid
            butterflies and their host plants. <em>Ecology Letters</em>,
            <em>n/a</em>(n/a). <a
            href="https://doi.org/10.1111/ele.13842">https://doi.org/10.1111/ele.13842</a>
            </div>
            <div id="ref-BramonMora2018Identifying" class="csl-entry"
            role="doc-biblioentry">
            Bramon Mora, B., Gravel, D., Gilarranz, L. J., Poisot, T.,
            &amp; Stouffer, D. B. (2018). Identifying a common backbone
            of interactions underlying food webs from different
            ecosystems. <em>Nature Communications</em>, <em>9</em>(1),
            2603. <a
            href="https://doi.org/10.1038/s41467-018-05056-0">https://doi.org/10.1038/s41467-018-05056-0</a>
            </div>
            <div id="ref-Buxton2021Key" class="csl-entry"
            role="doc-biblioentry">
            Buxton, R. T., Bennett, J. R., Reid, A. J., Shulman, C.,
            Cooke, S. J., Francis, C. M., Nyboer, E. A., Pritchard, G.,
            Binley, A. D., Avery-Gomm, S., Ban, N. C., Beazley, K. F.,
            Bennett, E., Blight, L. K., Bortolotti, L. E., Camfield, A.
            F., Gadallah, F., Jacob, A. L., Naujokaitis-Lewis, I., …
            Smith, P. A. (2021). Key information needs to move from
            knowledge to action for biodiversity conservation in Canada.
            <em>Biological Conservation</em>, <em>256</em>, 108983. <a
            href="https://doi.org/10.1016/j.biocon.2021.108983">https://doi.org/10.1016/j.biocon.2021.108983</a>
            </div>
            <div id="ref-Catchen2023Missing" class="csl-entry"
            role="doc-biblioentry">
            Catchen, M., Poisot, T., Pollock, L., &amp; Gonzalez, A.
            (2023). <em>The missing link: Discerning true from false
            negatives when sampling species interaction networks</em>
            (Preprint No. 4929). EcoEvoRXiV. <a
            href="https://doi.org/10.32942/X2DW22">https://doi.org/10.32942/X2DW22</a>
            </div>
            <div id="ref-Cazelles2016Theory" class="csl-entry"
            role="doc-biblioentry">
            Cazelles, K., Araújo, M. B., Mouquet, N., &amp; Gravel, D.
            (2016). A theory for species co-occurrence in interaction
            networks. <em>Theoretical Ecology</em>, <em>9</em>(1),
            39–48. <a
            href="https://doi.org/10.1007/s12080-015-0281-9">https://doi.org/10.1007/s12080-015-0281-9</a>
            </div>
            <div id="ref-Chami2022Machine" class="csl-entry"
            role="doc-biblioentry">
            Chami, I., Abu-El-Haija, S., Perozzi, B., Ré, C., &amp;
            Murphy, K. (2022). Machine Learning on Graphs: A Model and
            Comprehensive Taxonomy. <em>Journal of Machine Learning
            Research</em>, <em>23</em>(89), 1–64. <a
            href="http://jmlr.org/papers/v23/20-852.html">http://jmlr.org/papers/v23/20-852.html</a>
            </div>
            <div id="ref-Chen2017Deep" class="csl-entry"
            role="doc-biblioentry">
            Chen, D., Xue, Y., Fink, D., Chen, S., &amp; Gomes, C. P.
            (2017). <em>Deep Multi-species Embedding</em>. 3639–3646. <a
            href="https://www.ijcai.org/proceedings/2017/509">https://www.ijcai.org/proceedings/2017/509</a>
            </div>
            <div id="ref-Chen2017Harp" class="csl-entry"
            role="doc-biblioentry">
            Chen, H., Perozzi, B., Hu, Y., &amp; Skiena, S. (2017).
            <em>HARP: Hierarchical Representation Learning for
            Networks</em>. <a
            href="http://arxiv.org/abs/1706.07845">http://arxiv.org/abs/1706.07845</a>
            </div>
            <div id="ref-Choudry2013Saving" class="csl-entry"
            role="doc-biblioentry">
            Choudry, A. (2013). Saving biodiversity, for whom and for
            what? Conservation NGOs, complicity, colonialism and
            conquest in an era of capitalist globalization. In
            <em>NGOization: Complicity, contradictions and
            prospects</em> (pp. 24–44). Bloomsbury Publishing.
            </div>
            <div id="ref-Cieslak2020Tdistributed" class="csl-entry"
            role="doc-biblioentry">
            Cieslak, M. C., Castelfranco, A. M., Roncalli, V., Lenz, P.
            H., &amp; Hartline, D. K. (2020). T-Distributed Stochastic
            Neighbor Embedding (t-SNE): A tool for eco-physiological
            transcriptomic analysis. <em>Marine Genomics</em>,
            <em>51</em>, 100723. <a
            href="https://doi.org/10.1016/j.margen.2019.100723">https://doi.org/10.1016/j.margen.2019.100723</a>
            </div>
            <div id="ref-Csermely2004Strong" class="csl-entry"
            role="doc-biblioentry">
            Csermely, P. (2004). Strong links are important, but weak
            links stabilize them. <em>Trends in Biochemical
            Sciences</em>, <em>29</em>(7), 331–334. <a
            href="https://doi.org/10.1016/j.tibs.2004.05.004">https://doi.org/10.1016/j.tibs.2004.05.004</a>
            </div>
            <div id="ref-DallaRiva2016Exploring" class="csl-entry"
            role="doc-biblioentry">
            Dalla Riva, G. V., &amp; Stouffer, D. B. (2016). Exploring
            the evolutionary signature of food webs’ backbones using
            functional traits. <em>Oikos</em>, <em>125</em>(4), 446–456.
            <a
            href="https://doi.org/10.1111/oik.02305">https://doi.org/10.1111/oik.02305</a>
            </div>
            <div id="ref-Dallas2017Predicting" class="csl-entry"
            role="doc-biblioentry">
            Dallas, T., Park, A. W., &amp; Drake, J. M. (2017).
            Predicting cryptic links in host-parasite networks. <em>PLOS
            Computational Biology</em>, <em>13</em>(5), e1005557. <a
            href="https://doi.org/10.1371/journal.pcbi.1005557">https://doi.org/10.1371/journal.pcbi.1005557</a>
            </div>
            <div id="ref-Danisch2021Makie" class="csl-entry"
            role="doc-biblioentry">
            Danisch, S., &amp; Krumbiegel, J. (2021). Makie.jl: Flexible
            high-performance data visualization for Julia. <em>Journal
            of Open Source Software</em>, <em>6</em>(65), 3349. <a
            href="https://doi.org/10.21105/joss.03349">https://doi.org/10.21105/joss.03349</a>
            </div>
            <div id="ref-Dominguez2020Decolonising" class="csl-entry"
            role="doc-biblioentry">
            Domínguez, L., &amp; Luoma, C. (2020). Decolonising
            Conservation Policy: How Colonial Land and Conservation
            Ideologies Persist and Perpetuate Indigenous Injustices at
            the Expense of the Environment. <em>Land</em>, <em>9</em>(3,
            3), 65. <a
            href="https://doi.org/10.3390/land9030065">https://doi.org/10.3390/land9030065</a>
            </div>
            <div id="ref-Dunne2006Network" class="csl-entry"
            role="doc-biblioentry">
            Dunne, J. A. (2006). The Network Structure of Food Webs. In
            J. A. Dunne &amp; M. Pascual (Eds.), <em>Ecological
            networks: Linking structure and dynamics</em> (pp. 27–86).
            Oxford University Press.
            </div>
            <div id="ref-Eero2021Use" class="csl-entry"
            role="doc-biblioentry">
            Eero, M., Dierking, J., Humborg, C., Undeman, E., MacKenzie,
            B. R., Ojaveer, H., Salo, T., &amp; Köster, F. W. (2021).
            Use of food web knowledge in environmental conservation and
            management of living resources in the Baltic Sea. <em>ICES
            Journal of Marine Science</em>, <em>78</em>(8), 2645–2663.
            <a
            href="https://doi.org/10.1093/icesjms/fsab145">https://doi.org/10.1093/icesjms/fsab145</a>
            </div>
            <div id="ref-Eichhorn2019Steps" class="csl-entry"
            role="doc-biblioentry">
            Eichhorn, M. P., Baker, K., &amp; Griffiths, M. (2019).
            Steps towards decolonising biogeography. <em>Frontiers of
            Biogeography</em>, <em>12</em>(1), 1–7. <a
            href="https://doi.org/10.21425/F5FBG44795">https://doi.org/10.21425/F5FBG44795</a>
            </div>
            <div id="ref-Eklof2013Dimensionality" class="csl-entry"
            role="doc-biblioentry">
            Eklöf, A., Jacob, U., Kopp, J., Bosch, J., Castro-Urgal, R.,
            Chacoff, N. P., Dalsgaard, B., de Sassi, C., Galetti, M.,
            Guimarães, P. R., Lomáscolo, S. B., Martín González, A. M.,
            Pizo, M. A., Rader, R., Rodrigo, A., Tylianakis, J. M.,
            Vázquez, D. P., &amp; Allesina, S. (2013). The
            dimensionality of ecological networks. <em>Ecology
            Letters</em>, <em>16</em>(5), 577–583. <a
            href="https://doi.org/10.1111/ele.12081">https://doi.org/10.1111/ele.12081</a>
            </div>
            <div id="ref-Eklof2016Phylogenetic" class="csl-entry"
            role="doc-biblioentry">
            Eklöf, A., &amp; Stouffer, D. B. (2016). The phylogenetic
            component of food web structure and intervality.
            <em>Theoretical Ecology</em>, <em>9</em>(1), 107–115. <a
            href="https://doi.org/10.1007/s12080-015-0273-9">https://doi.org/10.1007/s12080-015-0273-9</a>
            </div>
            <div id="ref-Fortin2021Network" class="csl-entry"
            role="doc-biblioentry">
            Fortin, M.-J., Dale, M. R. T., &amp; Brimacombe, C. (2021).
            Network ecology in dynamic landscapes. <em>Proceedings of
            the Royal Society B: Biological Sciences</em>,
            <em>288</em>(1949), rspb.2020.1889, 20201889. <a
            href="https://doi.org/10.1098/rspb.2020.1889">https://doi.org/10.1098/rspb.2020.1889</a>
            </div>
            <div id="ref-Fricke2022Effects" class="csl-entry"
            role="doc-biblioentry">
            Fricke, E. C., Ordonez, A., Rogers, H. S., &amp; Svenning,
            J.-C. (2022). The effects of defaunation on plants’ capacity
            to track climate change. <em>Science</em>. <a
            href="https://www.science.org/doi/abs/10.1126/science.abk3510">https://www.science.org/doi/abs/10.1126/science.abk3510</a>
            </div>
            <div id="ref-Galiana2022Ecological" class="csl-entry"
            role="doc-biblioentry">
            Galiana, N., Lurgi, M., Bastazini, V. A. G., Bosch, J.,
            Cagnolo, L., Cazelles, K., Claramunt-López, B., Emer, C.,
            Fortin, M.-J., Grass, I., Hernández-Castellano, C., Jauker,
            F., Leroux, S. J., McCann, K., McLeod, A. M., Montoya, D.,
            Mulder, C., Osorio-Canadas, S., Reverté, S., … Montoya, J.
            M. (2022). Ecological network complexity scales with area.
            <em>Nature Ecology &amp; Evolution</em>, 1–8. <a
            href="https://doi.org/10.1038/s41559-021-01644-4">https://doi.org/10.1038/s41559-021-01644-4</a>
            </div>
            <div id="ref-Galiana2018Spatial" class="csl-entry"
            role="doc-biblioentry">
            Galiana, N., Lurgi, M., Claramunt-López, B., Fortin, M.-J.,
            Leroux, S., Cazelles, K., Gravel, D., &amp; Montoya, J. M.
            (2018). The spatial scaling of species interaction networks.
            <em>Nature Ecology &amp; Evolution</em>, <em>2</em>(5),
            782–790. <a
            href="https://doi.org/10.1038/s41559-018-0517-3">https://doi.org/10.1038/s41559-018-0517-3</a>
            </div>
            <div id="ref-Gaucher2021Outlier" class="csl-entry"
            role="doc-biblioentry">
            Gaucher, S., Klopp, O., &amp; Robin, G. (2021). Outlier
            detection in networks with missing links. <em>Computational
            Statistics &amp; Data Analysis</em>, <em>164</em>, 107308.
            <a
            href="https://doi.org/10.1016/j.csda.2021.107308">https://doi.org/10.1016/j.csda.2021.107308</a>
            </div>
            <div id="ref-Gibb2021Data" class="csl-entry"
            role="doc-biblioentry">
            Gibb, R., Albery, G. F., Becker, D. J., Brierley, L.,
            Connor, R., Dallas, T. A., Eskew, E. A., Farrell, M. J.,
            Rasmussen, A. L., Ryan, S. J., Sweeny, A., Carlson, C. J.,
            &amp; Poisot, T. (2021). Data Proliferation, Reconciliation,
            and Synthesis in Viral Ecology. <em>BioScience</em>,
            <em>71</em>(11), 1148–1156. <a
            href="https://doi.org/10.1093/biosci/biab080">https://doi.org/10.1093/biosci/biab080</a>
            </div>
            <div id="ref-Goebel2023Body" class="csl-entry"
            role="doc-biblioentry">
            Goebel, L. G. A., Vitorino, B. D., Frota, A. V. B., &amp;
            Santos-Filho, M. dos. (2023). Body mass determines the role
            of mammal species in a frugivore-large fruit interaction
            network in a Neotropical savanna. <em>Journal of Tropical
            Ecology</em>, <em>39</em>, e12. <a
            href="https://doi.org/10.1017/S0266467422000505">https://doi.org/10.1017/S0266467422000505</a>
            </div>
            <div id="ref-Goodfellow2016Deep" class="csl-entry"
            role="doc-biblioentry">
            Goodfellow, I., Bengio, Y., &amp; Courville, A. (2016).
            <em>Deep learning</em>. MIT Press.
            </div>
            <div id="ref-Goyal2018Graph" class="csl-entry"
            role="doc-biblioentry">
            Goyal, P., &amp; Ferrara, E. (2018). Graph embedding
            techniques, applications, and performance: A survey.
            <em>Knowledge-Based Systems</em>, <em>151</em>, 78–94. <a
            href="https://doi.org/10.1016/j.knosys.2018.03.022">https://doi.org/10.1016/j.knosys.2018.03.022</a>
            </div>
            <div id="ref-Gravel2018Bringing" class="csl-entry"
            role="doc-biblioentry">
            Gravel, D., Baiser, B., Dunne, J. A., Kopelke, J.-P.,
            Martinez, N. D., Nyman, T., Poisot, T., Stouffer, D. B.,
            Tylianakis, J. M., Wood, S. A., &amp; Roslin, T. (2018).
            Bringing Elton and Grinnell together: A quantitative
            framework to represent the biogeography of ecological
            interaction networks. <em>Ecography</em>, <em>0</em>(0). <a
            href="https://doi.org/10.1111/ecog.04006">https://doi.org/10.1111/ecog.04006</a>
            </div>
            <div id="ref-Gravel2013Inferring" class="csl-entry"
            role="doc-biblioentry">
            Gravel, D., Poisot, T., Albouy, C., Velez, L., &amp;
            Mouillot, D. (2013). Inferring food web structure from
            predator-prey body size relationships. <em>Methods in
            Ecology and Evolution</em>, <em>4</em>(11), 1083–1090. <a
            href="https://doi.org/10.1111/2041-210X.12103">https://doi.org/10.1111/2041-210X.12103</a>
            </div>
            <div id="ref-Green2019Disparate" class="csl-entry"
            role="doc-biblioentry">
            Green, B., &amp; Chen, Y. (2019). Disparate Interactions: An
            Algorithm-in-the-Loop Analysis of Fairness in Risk
            Assessments. <em>Proceedings of the Conference on Fairness,
            Accountability, and Transparency</em>, 90–99. <a
            href="https://doi.org/10.1145/3287560.3287563">https://doi.org/10.1145/3287560.3287563</a>
            </div>
            <div id="ref-Grover2016Node2vec" class="csl-entry"
            role="doc-biblioentry">
            Grover, A., &amp; Leskovec, J. (2016). Node2vec: Scalable
            Feature Learning for Networks. <em>Proceedings of the 22nd
            ACM SIGKDD International Conference on Knowledge Discovery
            and Data Mining</em>, 855–864. <a
            href="https://doi.org/10.1145/2939672.2939754">https://doi.org/10.1145/2939672.2939754</a>
            </div>
            <div id="ref-Grunig2020Crop" class="csl-entry"
            role="doc-biblioentry">
            Grünig, M., Mazzi, D., Calanca, P., Karger, D. N., &amp;
            Pellissier, L. (2020). Crop and forest pest metawebs shift
            towards increased linkage and suitability overlap under
            climate change. <em>Communications Biology</em>,
            <em>3</em>(1, 1), 1–10. <a
            href="https://doi.org/10.1038/s42003-020-0962-9">https://doi.org/10.1038/s42003-020-0962-9</a>
            </div>
            <div id="ref-Gupta2022Simultaneously" class="csl-entry"
            role="doc-biblioentry">
            Gupta, A., Furrer, R., &amp; Petchey, O. L. (2022).
            Simultaneously estimating food web connectance and structure
            with uncertainty. <em>Ecology and Evolution</em>,
            <em>12</em>(3), e8643. <a
            href="https://doi.org/10.1002/ece3.8643">https://doi.org/10.1002/ece3.8643</a>
            </div>
            <div id="ref-Gupta2021Graph" class="csl-entry"
            role="doc-biblioentry">
            Gupta, A., Matta, P., &amp; Pant, B. (2021). Graph neural
            network: Current state of Art, challenges and applications.
            <em>Materials Today: Proceedings</em>, <em>46</em>,
            10927–10932. <a
            href="https://doi.org/10.1016/j.matpr.2021.01.950">https://doi.org/10.1016/j.matpr.2021.01.950</a>
            </div>
            <div id="ref-Hadfield2014Tale" class="csl-entry"
            role="doc-biblioentry">
            Hadfield, J. D., Krasnov, B. R., Poulin, R., &amp; Nakagawa,
            S. (2014). A Tale of Two Phylogenies: Comparative Analyses
            of Ecological Interactions. <em>The American
            Naturalist</em>, <em>183</em>(2), 174–187. <a
            href="https://doi.org/10.1086/674445">https://doi.org/10.1086/674445</a>
            </div>
            <div id="ref-Herbert1965Dune" class="csl-entry"
            role="doc-biblioentry">
            Herbert, F. (1965). <em>Dune</em> (1st ed.). Chilton Book
            Company.
            </div>
            <div id="ref-Hinton2002Stochastic" class="csl-entry"
            role="doc-biblioentry">
            Hinton, G., &amp; Roweis, S. T. (2002). Stochastic neighbor
            embedding. <em>NIPS</em>, <em>15</em>, 833–840.
            </div>
            <div id="ref-Hoffmann2019Machine" class="csl-entry"
            role="doc-biblioentry">
            Hoffmann, J., Bar-Sinai, Y., Lee, L. M., Andrejevic, J.,
            Mishra, S., Rubinstein, S. M., &amp; Rycroft, C. H. (2019).
            Machine learning in a data-limited regime: Augmenting
            experiments with synthetic data uncovers order in crumpled
            sheets. <em>Science Advances</em>, <em>5</em>(4), eaau6792.
            <a
            href="https://doi.org/10.1126/sciadv.aau6792">https://doi.org/10.1126/sciadv.aau6792</a>
            </div>
            <div id="ref-Hortal2015Seven" class="csl-entry"
            role="doc-biblioentry">
            Hortal, J., de Bello, F., Diniz-Filho, J. A. F., Lewinsohn,
            T. M., Lobo, J. M., &amp; Ladle, R. J. (2015). Seven
            Shortfalls that Beset Large-Scale Knowledge of Biodiversity.
            <em>Annual Review of Ecology, Evolution, and
            Systematics</em>, <em>46</em>(1), 523–549. <a
            href="https://doi.org/10.1146/annurev-ecolsys-112414-054400">https://doi.org/10.1146/annurev-ecolsys-112414-054400</a>
            </div>
            <div id="ref-Jones2009Pantheria" class="csl-entry"
            role="doc-biblioentry">
            Jones, K. E., Bielby, J., Cardillo, M., Fritz, S. A.,
            O’Dell, J., Orme, C. D. L., Safi, K., Sechrest, W., Boakes,
            E. H., Carbone, C., Connolly, C., Cutts, M. J., Foster, J.
            K., Grenyer, R., Habib, M., Plaster, C. A., Price, S. A.,
            Rigby, E. A., Rist, J., … Purvis, A. (2009). PanTHERIA: A
            species‐level database of life history, ecology, and
            geography of extant and recently extinct mammals: Ecological
            Archives E090‐184. <em>Ecology</em>, <em>90</em>(9),
            2648–2648. <a
            href="https://doi.org/10.1890/08-1494.1">https://doi.org/10.1890/08-1494.1</a>
            </div>
            <div id="ref-Jordano2016Sampling" class="csl-entry"
            role="doc-biblioentry">
            Jordano, P. (2016). Sampling networks of ecological
            interactions. <em>Functional Ecology</em>, <em>30</em>(12),
            1883–1893. <a
            href="https://doi.org/10.1111/1365-2435.12763">https://doi.org/10.1111/1365-2435.12763</a>
            </div>
            <div id="ref-Lamba2019Deep" class="csl-entry"
            role="doc-biblioentry">
            Lamba, A., Cassey, P., Segaran, R. R., &amp; Koh, L. P.
            (2019). Deep learning for environmental conservation.
            <em>Current Biology</em>, <em>29</em>(19), R977–R982. <a
            href="https://doi.org/10.1016/j.cub.2019.08.016">https://doi.org/10.1016/j.cub.2019.08.016</a>
            </div>
            <div id="ref-Lenzner2022Naturalized" class="csl-entry"
            role="doc-biblioentry">
            Lenzner, B., Latombe, G., Schertler, A., Seebens, H., Yang,
            Q., Winter, M., Weigelt, P., van Kleunen, M., Pyšek, P.,
            Pergl, J., Kreft, H., Dawson, W., Dullinger, S., &amp; Essl,
            F. (2022). Naturalized alien floras still carry the legacy
            of European colonialism. <em>Nature Ecology &amp;
            Evolution</em>, 1–10. <a
            href="https://doi.org/10.1038/s41559-022-01865-1">https://doi.org/10.1038/s41559-022-01865-1</a>
            </div>
            <div id="ref-Llewelyn2022Predicting" class="csl-entry"
            role="doc-biblioentry">
            Llewelyn, J., Strona, G., Dickman, C. R., Greenville, A. C.,
            Wardle, G. M., Lee, M. S. Y., Doherty, S., Shabani, F.,
            Saltré, F., &amp; Bradshaw, C. J. A. (2022). <em>Predicting
            predator-prey interactions in terrestrial endotherms using
            random forest</em> [Preprint]. Ecology. <a
            href="https://doi.org/10.1101/2022.09.02.506446">https://doi.org/10.1101/2022.09.02.506446</a>
            </div>
            <div id="ref-Maaten2009Learning" class="csl-entry"
            role="doc-biblioentry">
            Maaten, L. van der. (2009). Learning a Parametric Embedding
            by Preserving Local Structure. <em>Proceedings of the Twelth
            International Conference on Artificial Intelligence and
            Statistics</em>, 384–391. <a
            href="https://proceedings.mlr.press/v5/maaten09a.html">https://proceedings.mlr.press/v5/maaten09a.html</a>
            </div>
            <div id="ref-Machen2021Thinking" class="csl-entry"
            role="doc-biblioentry">
            Machen, R., &amp; Nost, E. (2021). Thinking algorithmically:
            The making of hegemonic knowledge in climate governance.
            <em>Transactions of the Institute of British
            Geographers</em>, <em>46</em>(3), 555–569. <a
            href="https://doi.org/10.1111/tran.12441">https://doi.org/10.1111/tran.12441</a>
            </div>
            <div id="ref-Malaterre2019Functional" class="csl-entry"
            role="doc-biblioentry">
            Malaterre, C., Dussault, A. C., Mermans, E., Barker, G.,
            Beisner, B. E., Bouchard, F., Desjardins, E., Handa, I. T.,
            Kembel, S. W., Lajoie, G., Maris, V., Munson, A. D.,
            Odenbaugh, J., Poisot, T., Shapiro, B. J., &amp; Suttle, C.
            A. (2019). Functional Diversity: An Epistemic Roadmap.
            <em>BioScience</em>, <em>69</em>(10), 800–811. <a
            href="https://doi.org/10.1093/biosci/biz089">https://doi.org/10.1093/biosci/biz089</a>
            </div>
            <div id="ref-Martins2022Global" class="csl-entry"
            role="doc-biblioentry">
            Martins, L. P., Stouffer, D. B., Blendinger, P. G.,
            Böhning-Gaese, K., Buitrón-Jurado, G., Correia, M., Costa,
            J. M., Dehling, D. M., Donatti, C. I., Emer, C., Galetti,
            M., Heleno, R., Jordano, P., Menezes, Í., Morante-Filho, J.
            C., Muñoz, M. C., Neuschulz, E. L., Pizo, M. A., Quitián,
            M., … Tylianakis, J. M. (2022). Global and regional
            ecological boundaries explain abrupt spatial discontinuities
            in avian frugivory interactions. <em>Nature
            Communications</em>, <em>13</em>(1, 1), 6943. <a
            href="https://doi.org/10.1038/s41467-022-34355-w">https://doi.org/10.1038/s41467-022-34355-w</a>
            </div>
            <div id="ref-McLeod2021Sampling" class="csl-entry"
            role="doc-biblioentry">
            McLeod, A., Leroux, S. J., Gravel, D., Chu, C., Cirtwill, A.
            R., Fortin, M.-J., Galiana, N., Poisot, T., &amp; Wood, S.
            A. (2021). Sampling and asymptotic network properties of
            spatial multi-trophic networks. <em>Oikos</em>,
            <em>n/a</em>(n/a). <a
            href="https://doi.org/10.1111/oik.08650">https://doi.org/10.1111/oik.08650</a>
            </div>
            <div id="ref-Melnyk2020Graphkke" class="csl-entry"
            role="doc-biblioentry">
            Melnyk, K., Klus, S., Montavon, G., &amp; Conrad, T. O. F.
            (2020). GraphKKE: Graph Kernel Koopman embedding for human
            microbiome analysis. <em>Applied Network Science</em>,
            <em>5</em>(1), 96. <a
            href="https://doi.org/10.1007/s41109-020-00339-2">https://doi.org/10.1007/s41109-020-00339-2</a>
            </div>
            <div id="ref-Morales-Castilla2015Inferring"
            class="csl-entry" role="doc-biblioentry">
            Morales-Castilla, I., Matias, M. G., Gravel, D., &amp;
            Araújo, M. B. (2015). Inferring biotic interactions from
            proxies. <em>Trends in Ecology &amp; Evolution</em>,
            <em>30</em>(6), 347–356. <a
            href="https://doi.org/10.1016/j.tree.2015.03.014">https://doi.org/10.1016/j.tree.2015.03.014</a>
            </div>
            <div id="ref-Morand1998Density" class="csl-entry"
            role="doc-biblioentry">
            Morand, S., &amp; Poulin, R. (1998). Density, body mass and
            parasite species richness of terrestrial mammals.
            <em>Evolutionary Ecology</em>, <em>12</em>(6), 717–727. <a
            href="https://doi.org/10.1023/A:1006537600093">https://doi.org/10.1023/A:1006537600093</a>
            </div>
            <div id="ref-MoseboFernandes2020Machine" class="csl-entry"
            role="doc-biblioentry">
            Mosebo Fernandes, A. C., Quintero Gonzalez, R.,
            Lenihan-Clarke, M. A., Leslie Trotter, E. F., &amp; Jokar
            Arsanjani, J. (2020). Machine Learning for Conservation
            Planning in a Changing Climate. <em>Sustainability</em>,
            <em>12</em>(18, 18), 7657. <a
            href="https://doi.org/10.3390/su12187657">https://doi.org/10.3390/su12187657</a>
            </div>
            <div id="ref-Murphy2022Probabilistic" class="csl-entry"
            role="doc-biblioentry">
            Murphy, K. P. (2022). <em>Probabilistic machine learning: An
            introduction</em>. MIT Press. <a
            href="https://probml.ai">probml.ai</a>
            </div>
            <div id="ref-Naman2022Food" class="csl-entry"
            role="doc-biblioentry">
            Naman, S. M., White, S. M., Bellmore, J. R., McHugh, P. A.,
            Kaylor, M. J., Baxter, C. V., Danehy, R. J., Naiman, R. J.,
            &amp; Puls, A. L. (2022). Food web perspectives and methods
            for riverine fish conservation. <em>WIREs Water</em>,
            <em>n/a</em>(n/a), e1590. <a
            href="https://doi.org/10.1002/wat2.1590">https://doi.org/10.1002/wat2.1590</a>
            </div>
            <div id="ref-Narayanan2017Graph2vec" class="csl-entry"
            role="doc-biblioentry">
            Narayanan, A., Chandramohan, M., Venkatesan, R., Chen, L.,
            Liu, Y., &amp; Jaiswal, S. (2017). <em>Graph2vec: Learning
            Distributed Representations of Graphs</em>. <a
            href="http://arxiv.org/abs/1707.05005">http://arxiv.org/abs/1707.05005</a>
            </div>
            <div id="ref-Neutel2002Stability" class="csl-entry"
            role="doc-biblioentry">
            Neutel, A.-M., Heesterbeek, J. A. P., &amp; de Ruiter, P. C.
            (2002). Stability in Real Food Webs: Weak Links in Long
            Loops. <em>Science</em>, <em>296</em>(5570), 1120–1123. <a
            href="https://doi.org/10.1126/science.1068326">https://doi.org/10.1126/science.1068326</a>
            </div>
            <div id="ref-Nokmaq2021Awakening" class="csl-entry"
            role="doc-biblioentry">
            No’kmaq, M., Marshall, A., Beazley, K. F., Hum, J., joudry,
            shalan, Papadopoulos, A., Pictou, S., Rabesca, J., Young,
            L., &amp; Zurba, M. (2021). <span>“Awakening the sleeping
            giant”</span>: Re-Indigenization principles for transforming
            biodiversity conservation in Canada and beyond.
            <em>FACETS</em>, <em>6</em>(1), 839–869.
            </div>
            <div id="ref-Nost2021Political" class="csl-entry"
            role="doc-biblioentry">
            Nost, E., &amp; Goldstein, J. E. (2021). A political ecology
            of data. <em>Environment and Planning E: Nature and
            Space</em>, 25148486211043503. <a
            href="https://doi.org/10.1177/25148486211043503">https://doi.org/10.1177/25148486211043503</a>
            </div>
            <div id="ref-OConnor2020Unveiling" class="csl-entry"
            role="doc-biblioentry">
            O’Connor, L. M. J., Pollock, L. J., Braga, J., Ficetola, G.
            F., Maiorano, L., Martinez‐Almoyna, C., Montemaggiori, A.,
            Ohlmann, M., &amp; Thuiller, W. (2020). Unveiling the food
            webs of tetrapods across Europe through the prism of the
            Eltonian niche. <em>Journal of Biogeography</em>,
            <em>47</em>(1), 181–192. <a
            href="https://doi.org/10.1111/jbi.13773">https://doi.org/10.1111/jbi.13773</a>
            </div>
            <div id="ref-Pedersen2017Signatures" class="csl-entry"
            role="doc-biblioentry">
            Pedersen, E. J., Thompson, P. L., Ball, R. A., Fortin,
            M.-J., Gouhier, T. C., Link, H., Moritz, C., Nenzen, H.,
            Stanley, R. R. E., Taranu, Z. E., Gonzalez, A., Guichard,
            F., &amp; Pepin, P. (2017). Signatures of the collapse and
            incipient recovery of an overexploited marine ecosystem.
            <em>Royal Society Open Science</em>, <em>4</em>(7), 170215.
            <a
            href="https://doi.org/10.1098/rsos.170215">https://doi.org/10.1098/rsos.170215</a>
            </div>
            <div id="ref-Perozzi2014Deepwalk" class="csl-entry"
            role="doc-biblioentry">
            Perozzi, B., Al-Rfou, R., &amp; Skiena, S. (2014). DeepWalk:
            Online learning of social representations. <em>Proceedings
            of the 20th ACM SIGKDD International Conference on Knowledge
            Discovery and Data Mining</em>, 701–710. <a
            href="https://doi.org/10.1145/2623330.2623732">https://doi.org/10.1145/2623330.2623732</a>
            </div>
            <div id="ref-Pichler2020Machine" class="csl-entry"
            role="doc-biblioentry">
            Pichler, M., Boreux, V., Klein, A.-M., Schleuning, M., &amp;
            Hartig, F. (2020). Machine learning algorithms to infer
            trait-matching and predict species interactions in
            ecological networks. <em>Methods in Ecology and
            Evolution</em>, <em>11</em>(2), 281–293. <a
            href="https://doi.org/10.1111/2041-210X.13329">https://doi.org/10.1111/2041-210X.13329</a>
            </div>
            <div id="ref-Poisot2019Ecologicalnetworks" class="csl-entry"
            role="doc-biblioentry">
            Poisot, T., Belisle, Z., Hoebeke, L., Stock, M., &amp;
            Szefer, P. (2019). EcologicalNetworks.jl - analysing
            ecological networks. <em>Ecography</em>. <a
            href="https://doi.org/10.1111/ecog.04310">https://doi.org/10.1111/ecog.04310</a>
            </div>
            <div id="ref-Poisot2016Structure" class="csl-entry"
            role="doc-biblioentry">
            Poisot, T., Cirtwill, A. R., Cazelles, K., Gravel, D.,
            Fortin, M.-J., &amp; Stouffer, D. B. (2016). The structure
            of probabilistic networks. <em>Methods in Ecology and
            Evolution</em>, <em>7</em>(3), 303–312. <a
            href="https://doi.org/10.1111/2041-210X.12468">https://doi.org/10.1111/2041-210X.12468</a>
            </div>
            <div id="ref-Poisot2021Imputing" class="csl-entry"
            role="doc-biblioentry">
            Poisot, T., Ouellet, M.-A., Mollentze, N., Farrell, M. J.,
            Becker, D. J., Albery, G. F., Gibb, R. J., Seifert, S. N.,
            &amp; Carlson, C. J. (2021). <em>Imputing the mammalian
            virome with linear filtering and singular value
            decomposition</em>. <a
            href="http://arxiv.org/abs/2105.14973">http://arxiv.org/abs/2105.14973</a>
            </div>
            <div id="ref-Poisot2015Species" class="csl-entry"
            role="doc-biblioentry">
            Poisot, T., Stouffer, D. B., &amp; Gravel, D. (2015). Beyond
            species: Why ecological interaction networks vary through
            space and time. <em>Oikos</em>, <em>124</em>(3), 243–251. <a
            href="https://doi.org/10.1111/oik.01719">https://doi.org/10.1111/oik.01719</a>
            </div>
            <div id="ref-Raja2022Colonialism" class="csl-entry"
            role="doc-biblioentry">
            Raja, N. B. (2022). Colonialism shaped today’s biodiversity.
            <em>Nature Ecology &amp; Evolution</em>, 1–2. <a
            href="https://doi.org/10.1038/s41559-022-01903-y">https://doi.org/10.1038/s41559-022-01903-y</a>
            </div>
            <div id="ref-Ramasamy2015Compressive" class="csl-entry"
            role="doc-biblioentry">
            Ramasamy, D., &amp; Madhow, U. (2015). Compressive spectral
            embedding: Sidestepping the SVD. In C. Cortes, N. Lawrence,
            D. Lee, M. Sugiyama, &amp; R. Garnett (Eds.), <em>Advances
            in neural information processing systems</em> (Vol. 28).
            Curran Associates, Inc. <a
            href="https://proceedings.neurips.cc/paper/2015/file/4f6ffe13a5d75b2d6a3923922b3922e5-Paper.pdf">https://proceedings.neurips.cc/paper/2015/file/4f6ffe13a5d75b2d6a3923922b3922e5-Paper.pdf</a>
            </div>
            <div id="ref-Ray2021Biodiversity" class="csl-entry"
            role="doc-biblioentry">
            Ray, J. C., Grimm, J., &amp; Olive, A. (2021). The
            biodiversity crisis in Canada: Failures and challenges of
            federal and sub-national strategic and legal frameworks.
            <em>FACETS</em>, <em>6</em>, 1044–1068. <a
            href="https://doi.org/10.1139/facets-2020-0075">https://doi.org/10.1139/facets-2020-0075</a>
            </div>
            <div id="ref-Runghen2021Exploiting" class="csl-entry"
            role="doc-biblioentry">
            Runghen, R., Stouffer, D. B., &amp; Dalla Riva, G. V.
            (2021). <em>Exploiting node metadata to predict interactions
            in large networks using graph embedding and neural
            networks</em>. <a
            href="https://doi.org/10.1101/2021.06.10.447991">https://doi.org/10.1101/2021.06.10.447991</a>
            </div>
            <div id="ref-Saravia2021Ecological" class="csl-entry"
            role="doc-biblioentry">
            Saravia, L. A., Marina, T. I., Kristensen, N. P., De Troch,
            M., &amp; Momo, F. R. (2021). Ecological network assembly:
            How the regional metaweb influences local food webs.
            <em>Journal of Animal Ecology</em>, <em>n/a</em>(n/a). <a
            href="https://doi.org/10.1111/1365-2656.13652">https://doi.org/10.1111/1365-2656.13652</a>
            </div>
            <div id="ref-Schmidt2022Systemic" class="csl-entry"
            role="doc-biblioentry">
            Schmidt, C., &amp; Garroway, C. J. (2022). Systemic racism
            alters wildlife genetic diversity. <em>Proceedings of the
            National Academy of Sciences</em>, <em>119</em>(43),
            e2102860119. <a
            href="https://doi.org/10.1073/pnas.2102860119">https://doi.org/10.1073/pnas.2102860119</a>
            </div>
            <div id="ref-Stevenson2021Algorithmic" class="csl-entry"
            role="doc-biblioentry">
            Stevenson, M. T., &amp; Doleac, J. L. (2021).
            <em>Algorithmic Risk Assessment in the Hands of Humans</em>
            (SSRN Scholarly Paper No. 3489440). <a
            href="https://doi.org/10.2139/ssrn.3489440">https://doi.org/10.2139/ssrn.3489440</a>
            </div>
            <div id="ref-Stier2017Integrating" class="csl-entry"
            role="doc-biblioentry">
            Stier, A. C., Samhouri, J. F., Gray, S., Martone, R. G.,
            Mach, M. E., Halpern, B. S., Kappel, C. V., Scarborough, C.,
            &amp; Levin, P. S. (2017). Integrating Expert Perceptions
            into Food Web Conservation and Management. <em>Conservation
            Letters</em>, <em>10</em>(1), 67–76. <a
            href="https://doi.org/10.1111/conl.12245">https://doi.org/10.1111/conl.12245</a>
            </div>
            <div id="ref-Stouffer2007Evidence" class="csl-entry"
            role="doc-biblioentry">
            Stouffer, D. B., Camacho, J., Jiang, W., &amp; Nunes Amaral,
            L. A. (2007). Evidence for the existence of a robust pattern
            of prey selection in food webs. <em>Proceedings of the Royal
            Society B: Biological Sciences</em>, <em>274</em>(1621),
            1931–1940. <a
            href="https://doi.org/10.1098/rspb.2007.0571">https://doi.org/10.1098/rspb.2007.0571</a>
            </div>
            <div id="ref-Stouffer2012Evolutionary" class="csl-entry"
            role="doc-biblioentry">
            Stouffer, D. B., Sales-Pardo, M., Sirer, M. I., &amp;
            Bascompte, J. (2012). Evolutionary Conservation of Species’
            Roles in Food Webs. <em>Science</em>, <em>335</em>(6075),
            1489–1492. <a
            href="https://doi.org/10.1126/science.1216556">https://doi.org/10.1126/science.1216556</a>
            </div>
            <div id="ref-Strydom2022Food" class="csl-entry"
            role="doc-biblioentry">
            Strydom, T., Bouskila, S., Banville, F., Barros, C., Caron,
            D., Farrell, M. J., Fortin, M.-J., Hemming, V., Mercier, B.,
            Pollock, L. J., Runghen, R., Dalla Riva, G. V., &amp;
            Poisot, T. (2022). Food web reconstruction through
            phylogenetic transfer of low-rank network representation.
            <em>Methods in Ecology and Evolution</em>,
            <em>n/a</em>(n/a). <a
            href="https://doi.org/10.1111/2041-210X.13835">https://doi.org/10.1111/2041-210X.13835</a>
            </div>
            <div id="ref-Strydom2021Roadmap" class="csl-entry"
            role="doc-biblioentry">
            Strydom, T., Catchen, M. D., Banville, F., Caron, D.,
            Dansereau, G., Desjardins-Proulx, P., Forero-Muñoz, N. R.,
            Higino, G., Mercier, B., Gonzalez, A., Gravel, D., Pollock,
            L., &amp; Poisot, T. (2021). A roadmap towards predicting
            species interaction networks (across space and time).
            <em>Philosophical Transactions of the Royal Society B:
            Biological Sciences</em>, <em>376</em>(1837), 20210063. <a
            href="https://doi.org/10.1098/rstb.2021.0063">https://doi.org/10.1098/rstb.2021.0063</a>
            </div>
            <div id="ref-Strydom2021Svd" class="csl-entry"
            role="doc-biblioentry">
            Strydom, T., Dalla Riva, G. V., &amp; Poisot, T. (2021). SVD
            Entropy Reveals the High Complexity of Ecological Networks.
            <em>Frontiers in Ecology and Evolution</em>, <em>9</em>. <a
            href="https://doi.org/10.3389/fevo.2021.623141">https://doi.org/10.3389/fevo.2021.623141</a>
            </div>
            <div id="ref-Surendran2013Graph" class="csl-entry"
            role="doc-biblioentry">
            Surendran, S. (2013). Graph Embedding and Dimensionality
            Reduction - A Survey. <em>International Journal of Computer
            Science &amp; Engineering Technology</em>, <em>4</em>(1). <a
            href="https://www.semanticscholar.org/paper/Graph-Embedding-and-Dimensionality-Reduction-A-Surendran/3f413d591e4b2b876e033eeb9390e232ad4826ca">https://www.semanticscholar.org/paper/Graph-Embedding-and-Dimensionality-Reduction-A-Surendran/3f413d591e4b2b876e033eeb9390e232ad4826ca</a>
            </div>
            <div id="ref-Tang2015Line" class="csl-entry"
            role="doc-biblioentry">
            Tang, J., Qu, M., Wang, M., Zhang, M., Yan, J., &amp; Mei,
            Q. (2015). LINE: Large-scale Information Network Embedding.
            <em>Proceedings of the 24th International Conference on
            World Wide Web</em>, 1067–1077. <a
            href="https://doi.org/10.1145/2736277.2741093">https://doi.org/10.1145/2736277.2741093</a>
            </div>
            <div id="ref-Thurman2019Testing" class="csl-entry"
            role="doc-biblioentry">
            Thurman, L. L., Barner, A. K., Garcia, T. S., &amp;
            Chestnut, T. (2019). Testing the link between species
            interactions and co-occurrence in a trophic network.
            <em>Ecography</em>, <em>0</em>. <a
            href="https://doi.org/10.1111/ecog.04360">https://doi.org/10.1111/ecog.04360</a>
            </div>
            <div id="ref-Torres2020Glee" class="csl-entry"
            role="doc-biblioentry">
            Torres, L., Chan, K. S., &amp; Eliassi-Rad, T. (2020). GLEE:
            Geometric Laplacian Eigenmap Embedding. <em>Journal of
            Complex Networks</em>, <em>8</em>(2), cnaa007. <a
            href="https://doi.org/10.1093/comnet/cnaa007">https://doi.org/10.1093/comnet/cnaa007</a>
            </div>
            <div id="ref-Turak2017Using" class="csl-entry"
            role="doc-biblioentry">
            Turak, E., Brazill-Boast, J., Cooney, T., Drielsma, M.,
            DelaCruz, J., Dunkerley, G., Fernandez, M., Ferrier, S.,
            Gill, M., Jones, H., Koen, T., Leys, J., McGeoch, M.,
            Mihoub, J.-B., Scanes, P., Schmeller, D., &amp; Williams, K.
            (2017). Using the essential biodiversity variables framework
            to measure biodiversity change at national scale.
            <em>Biological Conservation</em>, <em>213</em>, 264–271. <a
            href="https://doi.org/10.1016/j.biocon.2016.08.019">https://doi.org/10.1016/j.biocon.2016.08.019</a>
            </div>
            <div id="ref-Wang2016Structural" class="csl-entry"
            role="doc-biblioentry">
            Wang, D., Cui, P., &amp; Zhu, W. (2016). Structural Deep
            Network Embedding. <em>Proceedings of the 22nd ACM SIGKDD
            International Conference on Knowledge Discovery and Data
            Mining</em>, 1225–1234. <a
            href="https://doi.org/10.1145/2939672.2939753">https://doi.org/10.1145/2939672.2939753</a>
            </div>
            <div id="ref-Wang2021Joint" class="csl-entry"
            role="doc-biblioentry">
            Wang, S., Arroyo, J., Vogelstein, J. T., &amp; Priebe, C. E.
            (2021). Joint Embedding of Graphs. <em>IEEE Transactions on
            Pattern Analysis and Machine Intelligence</em>,
            <em>43</em>(4), 1324–1336. <a
            href="https://doi.org/10.1109/TPAMI.2019.2948619">https://doi.org/10.1109/TPAMI.2019.2948619</a>
            </div>
            <div id="ref-Wardeh2021Predicting" class="csl-entry"
            role="doc-biblioentry">
            Wardeh, M., Baylis, M., &amp; Blagrove, M. S. C. (2021).
            Predicting mammalian hosts in which novel coronaviruses can
            be generated. <em>Nature Communications</em>, <em>12</em>(1,
            1), 780. <a
            href="https://doi.org/10.1038/s41467-021-21034-5">https://doi.org/10.1038/s41467-021-21034-5</a>
            </div>
            <div id="ref-Weiskopf2022Increasing" class="csl-entry"
            role="doc-biblioentry">
            Weiskopf, S. R., Harmáčková, Z. V., Johnson, C. G.,
            Londoño-Murcia, M. C., Miller, B. W., Myers, B. J. E.,
            Pereira, L., Arce-Plata, M. I., Blanchard, J. L., Ferrier,
            S., Fulton, E. A., Harfoot, M., Isbell, F., Johnson, J. A.,
            Mori, A. S., Weng, E., &amp; Rosa, I. M. D. (2022).
            Increasing the uptake of ecological model results in policy
            decisions to improve biodiversity outcomes.
            <em>Environmental Modelling &amp; Software</em>,
            <em>149</em>, 105318. <a
            href="https://doi.org/10.1016/j.envsoft.2022.105318">https://doi.org/10.1016/j.envsoft.2022.105318</a>
            </div>
            <div id="ref-Williams2000Simple" class="csl-entry"
            role="doc-biblioentry">
            Williams, R. J., &amp; Martinez, N. D. (2000). Simple rules
            yield complex food webs. <em>Nature</em>,
            <em>404</em>(6774), 180–183. <a
            href="https://doi.org/10.1038/35004572">https://doi.org/10.1038/35004572</a>
            </div>
            <div id="ref-Wood2015Effects" class="csl-entry"
            role="doc-biblioentry">
            Wood, S. A., Russell, R., Hanson, D., Williams, R. J., &amp;
            Dunne, J. A. (2015). Effects of spatial scale of sampling on
            food web structure. <em>Ecology and Evolution</em>,
            <em>5</em>(17), 3769–3782. <a
            href="https://doi.org/10.1002/ece3.1640">https://doi.org/10.1002/ece3.1640</a>
            </div>
            <div id="ref-Wu2021Maximum" class="csl-entry"
            role="doc-biblioentry">
            Wu, D., Palmer, D. R., &amp; Deford, D. R. (2021).
            <em>Maximum a Posteriori Inference of Random Dot Product
            Graphs via Conic Programming</em> (No. arXiv:2101.02180).
            arXiv. <a
            href="http://arxiv.org/abs/2101.02180">http://arxiv.org/abs/2101.02180</a>
            </div>
            <div id="ref-Xu2021Understanding" class="csl-entry"
            role="doc-biblioentry">
            Xu, M. (2021). Understanding Graph Embedding Methods and
            Their Applications. <em>SIAM Review</em>, <em>63</em>(4),
            825–853. <a
            href="https://doi.org/10.1137/20M1386062">https://doi.org/10.1137/20M1386062</a>
            </div>
            <div id="ref-Yan2005Graph" class="csl-entry"
            role="doc-biblioentry">
            Yan, S., Xu, D., Zhang, B., &amp; Zhang, H.-J. (2005). Graph
            embedding: A general framework for dimensionality reduction.
            <em>2005 IEEE Computer Society Conference on Computer Vision
            and Pattern Recognition (CVPR’05)</em>, <em>2</em>, 830–837
            vol. 2. <a
            href="https://doi.org/10.1109/CVPR.2005.170">https://doi.org/10.1109/CVPR.2005.170</a>
            </div>
            <div id="ref-Young2007Random" class="csl-entry"
            role="doc-biblioentry">
            Young, S. J., &amp; Scheinerman, E. R. (2007). Random Dot
            Product Graph Models for Social Networks. In A. Bonato &amp;
            F. R. K. Chung (Eds.), <em>Algorithms and Models for the
            Web-Graph</em> (pp. 138–149). Springer. <a
            href="https://doi.org/10.1007/978-3-540-77004-6_11">https://doi.org/10.1007/978-3-540-77004-6_11</a>
            </div>
            <div id="ref-Zhou2020Graph" class="csl-entry"
            role="doc-biblioentry">
            Zhou, J., Cui, G., Hu, S., Zhang, Z., Yang, C., Liu, Z.,
            Wang, L., Li, C., &amp; Sun, M. (2020). Graph neural
            networks: A review of methods and applications. <em>AI
            Open</em>, <em>1</em>, 57–81. <a
            href="https://doi.org/10.1016/j.aiopen.2021.01.001">https://doi.org/10.1016/j.aiopen.2021.01.001</a>
            </div>
            </div>
        </article>
    </main>

    <footer>
        <div class="container">
            <h2>Affiliations</h2> 
            <sup>1</sup>&nbsp;Département de Sciences Biologiques,
            Université de Montréal, Montréal,
            Canada; <sup>2</sup>&nbsp;Quebec Centre for Biodiversity
            Science, Montréal, Canada; <sup>3</sup>&nbsp;Département de
            Biologie, Université de Sherbrooke, Sherbrooke,
            Canada; <sup>4</sup>&nbsp;Department of Forest Resources
            Management, University of British Columbia, Vancouver, B.C.,
            Canada; <sup>5</sup>&nbsp;Department of Biology, McGill
            University, Montréal, Canada; <sup>6</sup>&nbsp;Department
            of Ecology &amp; Evolutionary Biology, University of
            Toronto, Toronto, Canada; <sup>7</sup>&nbsp;Department of
            Forest and Conservation Sciences, University of British
            Columbia, Vancouver, Canada; <sup>8</sup>&nbsp;Centre for
            Integrative Ecology, School of Biological Sciences,
            University of Canterbury, Canterbury, New
            Zealand; <sup>9</sup>&nbsp;School of Mathematics and
            Statistics, University of Canterbury, Canterbury, New
            Zealand
            <br />
            
        </div>
    </footer>
    
</body>
</html>